<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Updated from Charlie George">
<meta name="dcterms.date" content="2025-11-10">

<title>Solution: Pathway and gene set enrichment analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="10-pathways_files/libs/clipboard/clipboard.min.js"></script>
<script src="10-pathways_files/libs/quarto-html/quarto.js"></script>
<script src="10-pathways_files/libs/quarto-html/popper.min.js"></script>
<script src="10-pathways_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="10-pathways_files/libs/quarto-html/anchor.min.js"></script>
<link href="10-pathways_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="10-pathways_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="10-pathways_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="10-pathways_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="10-pathways_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="10-pathways_files/libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<script src="10-pathways_files/libs/plotly-binding-4.10.2/plotly.js"></script>
<script src="10-pathways_files/libs/typedarray-0.1/typedarray.min.js"></script>
<script src="10-pathways_files/libs/jquery-3.5.1/jquery.min.js"></script>
<link href="10-pathways_files/libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="10-pathways_files/libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="10-pathways_files/libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="10-pathways_files/libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Solution: Pathway and gene set enrichment analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Updated from Charlie George </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 10, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<ul>
<li><p>We will use the <a href="https://cran.r-project.org/web/packages/gprofiler2/index.html"><strong>gprofiler2</strong></a> R package to perform an over-representation analysis (ORA) on the output table of differentially expressed genes from DESeq2.</p></li>
<li><p>The major aims of this exercise is to get you used to reading online tool documentation, playing with different parameters of the relevant functions and interpreting outputs.</p></li>
</ul>
<section id="ora" class="level1">
<h1>ORA</h1>
<section id="load-libraries." class="level3">
<h3 class="anchored" data-anchor-id="load-libraries.">1. Load libraries.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>) <span class="co"># For rendering </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gprofiler2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-output-table-of-differentially-expressed-genes-from-deseq2." class="level3">
<h3 class="anchored" data-anchor-id="read-in-output-table-of-differentially-expressed-genes-from-deseq2.">2. Read in output table of differentially expressed genes from DESeq2.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>all_genes_df <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"dds_results.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 19053 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (1): gene_id
dbl (6): baseMean, log2FoldChange, lfcSE, stat, pvalue, padj

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<ul>
<li>Check its dimensions and look at first few rows.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(all_genes_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 19053     7</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_genes_df)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 7
  gene_id            baseMean log2FoldChange  lfcSE   stat   pvalue    padj
  &lt;chr&gt;                 &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
1 ENSMUSG00000098104    478.         -0.270  0.0923 -2.93  0.00339  0.00905
2 ENSMUSG00000033845    690.         -0.160  0.0798 -2.01  0.0446   0.0898 
3 ENSMUSG00000025903   3003.         -0.219  0.0608 -3.61  0.000310 0.00101
4 ENSMUSG00000033813    889.          0.0555 0.0782  0.710 0.478    0.620  
5 ENSMUSG00000062588     18.4        -0.497  0.434  -1.15  0.252    0.383  
6 ENSMUSG00000033793   1106.         -0.201  0.0718 -2.80  0.00510  0.0131 </code></pre>
</div>
</div>
</section>
<section id="create-input-gene-lists." class="level3">
<h3 class="anchored" data-anchor-id="create-input-gene-lists.">3. Create input gene lists.</h3>
<ul>
<li>Create these 3 data frames by subsetting the original one:
<ul>
<li><strong>all_genes_tested_df</strong> object containing table of all genes tested in the differential expression analysis, which will be the background set</li>
<li><strong>upreg_genes_df</strong> object containing all significantly upregulated genes (‘padj’ &lt; 0.05 &amp; ‘log2FoldChange’ &gt; 1)</li>
<li><strong>downreg_genes_df</strong> object containing all significantly downregulated genes (‘padj’ &lt; 0.05 &amp; ‘log2FoldChange’ &lt; -1)</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>all_genes_tested_df <span class="ot">&lt;-</span> all_genes_df <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(padj))</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># P-values set to NA are those from genes with all counts equal to zero (https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#why-are-some-p-values-set-to-na)</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>upreg_genes_df <span class="ot">&lt;-</span> all_genes_tested_df <span class="sc">%&gt;%</span> </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                     dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>downreg_genes_df <span class="ot">&lt;-</span> all_genes_tested_df <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Check the dimensions of the data frames.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(all_genes_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 19053     7</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(all_genes_tested_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 17943     7</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(upreg_genes_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1493    7</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(downreg_genes_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1085    7</code></pre>
</div>
</div>
<ul>
<li>From each of the 3 data frames, get a vector of the Ensembl ids.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>all_gene_tested_ids <span class="ot">&lt;-</span> all_genes_tested_df <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene_id)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>upreg_gene_ids <span class="ot">&lt;-</span> upreg_genes_df <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene_id)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>downreg_gene_ids <span class="ot">&lt;-</span> downreg_genes_df <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="perform-ora-using-gprofiler2." class="level3">
<h3 class="anchored" data-anchor-id="perform-ora-using-gprofiler2.">4. Perform ORA using <strong>gprofiler2</strong>.</h3>
<ol type="a">
<li>First, we want to do ORA on upregulated genes (vs.&nbsp;all genes tested as background). What function from gprofiler2 should we use? What parameters of that function should we include or change? Look at the <a href="https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html">documentation</a>.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>?<span class="fu">gost</span>() <span class="co"># Get help for function</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Write ORA query</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>gost_results_obj <span class="ot">&lt;-</span> <span class="fu">gost</span>(<span class="at">query =</span> upreg_gene_ids, </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">organism =</span> <span class="st">"mmusculus"</span>, </span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ordered_query =</span> <span class="cn">FALSE</span>,    </span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">multi_query =</span> <span class="cn">FALSE</span>, </span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">significant =</span> <span class="cn">TRUE</span>, </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">exclude_iea =</span> <span class="cn">TRUE</span>, </span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">measure_underrepresentation =</span> <span class="cn">FALSE</span>, </span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">evcodes =</span> <span class="cn">FALSE</span>, </span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">user_threshold =</span> <span class="fl">0.05</span>, </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>                         <span class="at">correction_method =</span> <span class="st">"g_SCS"</span>,  </span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>                         <span class="at">domain_scope =</span> <span class="st">"custom_annotated"</span>,</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">custom_bg =</span> all_gene_tested_ids, </span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>                         <span class="at">numeric_ns =</span> <span class="st">""</span>, </span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sources =</span> <span class="cn">NULL</span>, </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">as_short_link =</span> <span class="cn">FALSE</span>,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>                         <span class="at">highlight =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                         </span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a><span class="co"># gost() parameters:</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="do">## `ordered_query` - Change to `TRUE` if input gene list is ranked; "testing is then performed iteratively, starting from the first gene and sequentially adding genes one by one" (kolberg_gprofiler2_2020)</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="do">## `exclude_iea` - Change to `TRUE` if we want to be stringent with gene sets considered i.e. only look at curated GO gene sets</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a><span class="do">## measure_underrepresentation - whether overlap is less than expected</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a><span class="do">## `correction_method = "g_SCS"` - This is their own multiple testing correction; more stringent than Bonferroni or Benjamini-Hochberg, takes into account the unevenly distributed structure of functionally annotated gene sets</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a><span class="do">## `domain_scope = "custom_annotated"` - "Use the set of genes that are annotated in the data source and are also included in the user provided background list (documentation)"</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a><span class="do">## `sources` - "We recommend that initial pathway enrichment analyses includes only biological processes (BPs) of GO and molecular pathways of Reactome. Large pathways are of limited interpretative value, whereas numerous small pathways decrease the statistical power because of excessive multiple testing. (doi: 10.1038/s41596-018-0103-9)"</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a><span class="co"># gost() output coulumns - See gprofiler2 documentation</span></span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="do">## query_size - could be different depending on the source of gene sets/pathways because it considers genes annotated in the source</span></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a><span class="do">## precision - "the proportion of genes in the input list that are annotated to the function (defined as intersection_size/query_size)"</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a><span class="do">## recall - "the proportion of functionally annotated genes that the query recovers (defined as intersection_size/term_size)"</span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a><span class="do">## effective_domain_size - Different per source when `domain_scope = "custom_annotated"` as in query_size because not all background genes are annotated in the source</span></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a><span class="do">## source_order - "numeric order for the term within its data source (this is important for drawing the results)"; closely related terms are closer to each other in the order</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>What type of object is the output? Explore contents.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(gost_results_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "list"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(gost_results_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       Length Class      Mode
result 14     data.frame list
meta    5     -none-     list</code></pre>
</div>
</div>
<ol start="2" type="a">
<li>Plot our results using a function from gprofiler2. Which one is it?</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gostplot</span>(gost_results_obj, <span class="at">capped =</span> <span class="cn">TRUE</span>, <span class="at">interactive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-3c4966fcba5550651de6" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-3c4966fcba5550651de6">{"x":{"data":[{"x":[62.159638134572553,75.762374654986715,75.281712587127558,49.993828217021566,76.045712294987894,62.162167934929712,51.939244691672577,52.387019354888743,49.781324987020675,77.282784669635944,59.703201987776538,76.076069899273733,59.822102604562758,77.285314469993082,43.714863730566684,76.078599699630885,52.384489554531584,46.596306337364467,50.236689051308304,49.783854787377827,49.811682591306521,59.738619192776689,50.287285058451367,92.16560017076975,75.208348376770104,76.453010152489611,49.976119614521494,56.133653683833018,67.947821351739663,49.809152790949362,77.029804633920591,67.950351152096829,107.13442888404676,91.262461443265963,71.861422504256069,43.836294147710042,59.505877559918574,75.681421043557805,93.802381001848019,67.963000153882604,79.908717440361244,91.264991243623115,69.417635359245836,59.852460208848598,76.096308302130964,68.155264981026249,49.978649414878646,90.842514583978485,75.549871424985824,90.478223332548396,50.879258342025274,71.866482104970387,75.600467432128895,64.590776277797033,65.473676602443589,65.5546302138725,43.512479701994401,107.39752812119073,60.057374037778025,60.831492947066991,82.271550973942567,76.607327974275975,50.254397653808368,86.847959820033168,50.874198741310977,107.40258772190502,75.602997232486047,77.027274833563439,50.256927454165528,90.144289685404118,68.327291405312693,76.050771895702212,75.939460679987448,92.168129971126902,83.106385091803205,76.316400933203326,75.757315054272411,60.059903838135185,60.457082494208279,86.005536301101074,78.092320783925047,53.899839968466502,43.709804129852373,65.5116236078009,43.83376434735289,71.593263666397803,59.872698611705822,77.287844270350249,49.417033735590579,68.446192022098899,69.490999569603275],"y":[17,15.336621779728503,13.412952046810675,12.376376781917749,11.702138346088722,10.774388309996146,9.9598249309555378,8.7229924852176612,8.5442148703601308,8.4708957775947926,8.4600019449909976,8.4062433167050479,8.4062433167050479,8.2060502989954269,7.79750401134031,7.5805604905910382,7.5370060042262548,7.4249302687416918,7.4168327212591354,7.2237359864308814,7.1934341482619919,6.1235988831261858,5.852314192075923,5.6357553535827316,5.5770847678200042,5.5519417686855856,5.4226676489692727,5.3270134990356297,5.2588912600100821,4.6743259489186295,4.6637035908267892,4.5654861363717805,4.5464879535712432,4.3405855555914412,4.2831779425815517,4.2056291610934711,4.0644954836548006,4.0383702326433717,4.0107938070502982,4.0040740479245835,3.9850568459474669,3.9461967490767362,3.9278551494360165,3.8948927167064613,3.8831559131422444,3.8598701710518024,3.8223546674174429,3.8223546674174429,3.7778063145907663,3.7136544367639575,3.6546530474573484,3.6546513421575018,3.6079214646647983,3.595352214746192,3.2837263637954899,3.2149206180457344,3.0887510652735473,2.9342228282757632,2.815784929174233,2.812153213543807,2.8081150751071515,2.7524836833077519,2.6644760776572634,2.4973806247212651,2.4160313313829733,2.4138134602322303,2.4082441900076419,2.3400208597968319,2.2910795770962467,2.2910795770962467,2.278025045600542,2.2766677503752799,2.2756089815593485,2.1584674132430255,2.1220899573934116,2.0154188406736386,1.8485581981036345,1.845186540655015,1.6879580841335329,1.6440999518520778,1.6296747832110996,1.5281985568046346,1.4736727744379268,1.4124466377366311,1.403910306167045,1.3955945294669672,1.3955945294669672,1.365647275498338,1.3432031477291331,1.3432031477291331,1.3345771281489549],"text":["GO:0032501 (4443) <br> multicellular organismal process <br> 6.352e-18","GO:0048731 (2609) <br> system development <br> 4.607e-16","GO:0048513 (1911) <br> animal organ development <br> 3.864e-14","GO:0007275 (3109) <br> multicellular organism development <br> 4.204e-13","GO:0048856 (3850) <br> anatomical structure development <br> 1.985e-12","GO:0032502 (4121) <br> developmental process <br> 1.681e-11","GO:0009653 (1806) <br> anatomical structure morphogenesis <br> 1.097e-10","GO:0009888 (1303) <br> tissue development <br> 1.892e-09","GO:0007154 (3732) <br> cell communication <br> 2.856e-09","GO:0051239 (2176) <br> regulation of multicellular organismal process <br> 3.381e-09","GO:0023052 (3726) <br> signaling <br> 3.467e-09","GO:0048869 (2809) <br> cellular developmental process <br> 3.924e-09","GO:0030154 (2809) <br> cell differentiation <br> 3.924e-09","GO:0051240 (1284) <br> positive regulation of multicellular organismal process <br> 6.222e-09","GO:0001503 (276) <br> ossification <br> 1.594e-08","GO:0048870 (1200) <br> cell motility <br> 2.627e-08","GO:0009887 (662) <br> animal organ morphogenesis <br> 2.904e-08","GO:0003008 (1086) <br> system process <br> 3.759e-08","GO:0007399 (1642) <br> nervous system development <br> 3.830e-08","GO:0007155 (868) <br> cell adhesion <br> 5.974e-08","GO:0007166 (1812) <br> cell surface receptor signaling pathway <br> 6.406e-08","GO:0030030 (1193) <br> cell projection organization <br> 7.523e-07","GO:0007423 (373) <br> sensory organ development <br> 1.405e-06","GO:0120036 (1174) <br> plasma membrane bounded cell projection organization <br> 2.313e-06","GO:0048468 (2116) <br> cell development <br> 2.648e-06","GO:0050793 (1802) <br> regulation of developmental process <br> 2.806e-06","GO:0007267 (784) <br> cell-cell signaling <br> 3.779e-06","GO:0016477 (1089) <br> cell migration <br> 4.710e-06","GO:0040011 (871) <br> locomotion <br> 5.509e-06","GO:0007165 (3410) <br> signal transduction <br> 2.117e-05","GO:0051094 (1007) <br> positive regulation of developmental process <br> 2.169e-05","GO:0040012 (766) <br> regulation of locomotion <br> 2.720e-05","GO:2000026 (1071) <br> regulation of multicellular organismal development <br> 2.841e-05","GO:0099536 (507) <br> synaptic signaling <br> 4.565e-05","GO:0045595 (1167) <br> regulation of cell differentiation <br> 5.210e-05","GO:0001654 (259) <br> eye development <br> 6.228e-05","GO:0022008 (1272) <br> neurogenesis <br> 8.620e-05","GO:0048699 (1088) <br> generation of neurons <br> 9.154e-05","GO:0150063 (262) <br> visual system development <br> 9.755e-05","GO:0040017 (445) <br> positive regulation of locomotion <br> 9.907e-05","GO:0060429 (774) <br> epithelium development <br> 1.035e-04","GO:0099537 (477) <br> trans-synaptic signaling <br> 1.132e-04","GO:0043010 (229) <br> camera-type eye development <br> 1.181e-04","GO:0030182 (1032) <br> neuron differentiation <br> 1.274e-04","GO:0048880 (264) <br> sensory system development <br> 1.309e-04","GO:0042127 (1182) <br> regulation of cell population proliferation <br> 1.381e-04","GO:0007268 (472) <br> chemical synaptic transmission <br> 1.505e-04","GO:0098916 (472) <br> anterograde trans-synaptic signaling <br> 1.505e-04","GO:0048646 (780) <br> anatomical structure formation involved in morphogenesis <br> 1.668e-04","GO:0098609 (593) <br> cell-cell adhesion <br> 1.934e-04","GO:0008285 (507) <br> negative regulation of cell population proliferation <br> 2.215e-04","GO:0045597 (684) <br> positive regulation of cell differentiation <br> 2.215e-04","GO:0048666 (878) <br> neuron development <br> 2.466e-04","GO:0034330 (540) <br> cell junction organization <br> 2.539e-04","GO:0035239 (596) <br> tube morphogenesis <br> 5.203e-04","GO:0035295 (738) <br> tube development <br> 6.096e-04","GO:0000902 (725) <br> cell morphogenesis <br> 8.152e-04","GO:2000145 (738) <br> regulation of cell motility <br> 1.164e-03","GO:0030334 (708) <br> regulation of cell migration <br> 1.528e-03","GO:0031175 (784) <br> neuron projection development <br> 1.541e-03","GO:0061564 (356) <br> axon development <br> 1.556e-03","GO:0050877 (635) <br> nervous system process <br> 1.768e-03","GO:0007409 (314) <br> axonogenesis <br> 2.165e-03","GO:0072359 (777) <br> circulatory system development <br> 3.181e-03","GO:0008283 (1438) <br> cell population proliferation <br> 3.837e-03","GO:2000147 (434) <br> positive regulation of cell motility <br> 3.856e-03","GO:0048667 (442) <br> cell morphogenesis involved in neuron differentiation <br> 3.906e-03","GO:0051093 (672) <br> negative regulation of developmental process <br> 4.571e-03","GO:0007411 (139) <br> axon guidance <br> 5.116e-03","GO:0097485 (139) <br> neuron projection guidance <br> 5.116e-03","GO:0042221 (2276) <br> response to chemical <br> 5.272e-03","GO:0048858 (493) <br> cell projection morphogenesis <br> 5.288e-03","GO:0048812 (477) <br> neuron projection morphogenesis <br> 5.301e-03","GO:0120039 (488) <br> plasma membrane bounded cell projection morphogenesis <br> 6.943e-03","GO:0065008 (1939) <br> regulation of biological quality <br> 7.549e-03","GO:0050673 (327) <br> epithelial cell proliferation <br> 9.651e-03","GO:0048729 (416) <br> tissue morphogenesis <br> 1.417e-02","GO:0030335 (424) <br> positive regulation of cell migration <br> 1.428e-02","GO:0030855 (412) <br> epithelial cell differentiation <br> 2.051e-02","GO:0072001 (202) <br> renal system development <br> 2.269e-02","GO:0051716 (4337) <br> cellular response to stimulus <br> 2.346e-02","GO:0010720 (408) <br> positive regulation of cell development <br> 2.963e-02","GO:0001501 (331) <br> skeletal system development <br> 3.360e-02","GO:0035272 (34) <br> exocrine system development <br> 3.869e-02","GO:0001649 (144) <br> osteoblast differentiation <br> 3.945e-02","GO:0045229 (192) <br> external encapsulating structure organization <br> 4.022e-02","GO:0030198 (192) <br> extracellular matrix organization <br> 4.022e-02","GO:0051241 (842) <br> negative regulation of multicellular organismal process <br> 4.309e-02","GO:0006935 (288) <br> chemotaxis <br> 4.537e-02","GO:0042330 (288) <br> taxis <br> 4.537e-02","GO:0043062 (193) <br> extracellular structure organization <br> 4.628e-02"],"key":["GO:0032501","GO:0048731","GO:0048513","GO:0007275","GO:0048856","GO:0032502","GO:0009653","GO:0009888","GO:0007154","GO:0051239","GO:0023052","GO:0048869","GO:0030154","GO:0051240","GO:0001503","GO:0048870","GO:0009887","GO:0003008","GO:0007399","GO:0007155","GO:0007166","GO:0030030","GO:0007423","GO:0120036","GO:0048468","GO:0050793","GO:0007267","GO:0016477","GO:0040011","GO:0007165","GO:0051094","GO:0040012","GO:2000026","GO:0099536","GO:0045595","GO:0001654","GO:0022008","GO:0048699","GO:0150063","GO:0040017","GO:0060429","GO:0099537","GO:0043010","GO:0030182","GO:0048880","GO:0042127","GO:0007268","GO:0098916","GO:0048646","GO:0098609","GO:0008285","GO:0045597","GO:0048666","GO:0034330","GO:0035239","GO:0035295","GO:0000902","GO:2000145","GO:0030334","GO:0031175","GO:0061564","GO:0050877","GO:0007409","GO:0072359","GO:0008283","GO:2000147","GO:0048667","GO:0051093","GO:0007411","GO:0097485","GO:0042221","GO:0048858","GO:0048812","GO:0120039","GO:0065008","GO:0050673","GO:0048729","GO:0030335","GO:0030855","GO:0072001","GO:0051716","GO:0010720","GO:0001501","GO:0035272","GO:0001649","GO:0045229","GO:0030198","GO:0051241","GO:0006935","GO:0042330","GO:0043062"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.80000000000000004,"size":[22.095044393964042,21.065595291859559,20.434082602011827,21.411284453140311,21.823800482978342,21.953100258546321,20.316869350465403,19.622818336014973,21.764319684461519,20.700355204705534,21.761239851078624,21.212056529276872,21.212056529276872,19.590871771701451,15.786285854749117,19.442883188215681,18.076082544635039,19.22195846010133,20.117512655941169,18.714156511391462,20.323771641274078,19.430020948045318,16.620731988973798,19.394670584408409,20.643378605298,20.312253464785375,18.477690973027336,19.228106358092081,18.722106721001474,21.590782733540944,19.052679900601216,18.423194346494352,19.190924617787029,17.418699722310961,19.381481887440675,15.602625065984768,19.570416639039269,19.226059214354727,15.636106885277737,17.085436902512729,18.4475964982825,17.263885431616249,15.238607515972017,19.107849354716695,15.658164048010281,19.409633935368124,17.236959058355662,17.236959058355662,18.46570689311762,17.808559557511707,17.418699722310961,18.154580931374287,18.740534956673521,17.576925704074707,17.820935765126428,18.335397097140174,18.293308239480449,18.335397097140174,18.236924556045079,18.477690973027336,16.495073744484998,17.975469913512335,16.150565769155275,18.456671970152541,19.835578849994551,17.020527291262969,17.067926058187329,18.112135439670144,13.62535144233572,13.62535144233572,20.791513148409948,17.347845715211314,17.263885431616249,17.321960027749867,20.464118939150438,16.262932392160565,16.909946036755073,16.959787212435252,16.884592389029979,14.855344879972034,22.049606435754676,16.858941516764062,16.296412811388283,3.7795275590551185,13.748135962630579,14.696411554466177,14.696411554466177,18.643896934518033,15.907685767928792,15.907685767928792,14.712779063920467],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData1f2373df","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[37.121612702342979,31.238586173070495,34.472353013677044,30.692034030996147,30.778065386693033,34.78611442857158,34.78358409458049,38.76379946255711,32.501222834621878,38.191943980571914,35.302302562752907,34.791175096553751,32.741604563774949,32.488571164666453,32.503753168612967,37.655513174461902,37.389828105397982,36.033569086176456,32.341811793183531,34.021953563263928,35.530032621950546,38.298218008197473,35.532562955941636],"y":[17,17,8.2260632847445994,7.9119604094445615,7.8279055842298035,7.6052840478214048,7.0653924929033893,6.1327947770761702,5.4079412591504035,5.2587762127029452,4.6139191908419948,4.4395607985362293,4.3305697254076563,4.2610502226644318,4.2565365619213749,4.1661795324505926,3.715612311093202,2.7640036490245441,2.5791944116820136,1.9201502931132151,1.850649100572195,1.4355238999297104,1.3069244560821554],"text":["GO:0071944 (3030) <br> cell periphery <br> 1.453e-25","GO:0005886 (2763) <br> plasma membrane <br> 9.715e-18","GO:0036477 (718) <br> somatodendritic compartment <br> 5.942e-09","GO:0005576 (917) <br> extracellular region <br> 1.225e-08","GO:0005615 (806) <br> extracellular space <br> 1.486e-08","GO:0043005 (933) <br> neuron projection <br> 2.482e-08","GO:0042995 (1514) <br> cell projection <br> 8.602e-08","GO:0120025 (1493) <br> plasma membrane bounded cell projection <br> 7.366e-07","GO:0030424 (485) <br> axon <br> 3.909e-06","GO:0098590 (773) <br> plasma membrane region <br> 5.511e-06","GO:0044297 (564) <br> cell body <br> 2.433e-05","GO:0043025 (490) <br> neuronal cell body <br> 3.634e-05","GO:0031012 (237) <br> extracellular matrix <br> 4.671e-05","GO:0030312 (238) <br> external encapsulating structure <br> 5.482e-05","GO:0030425 (494) <br> dendrite <br> 5.539e-05","GO:0097447 (496) <br> dendritic tree <br> 6.821e-05","GO:0097060 (260) <br> synaptic membrane <br> 1.925e-04","GO:0062023 (192) <br> collagen-containing extracellular matrix <br> 1.722e-03","GO:0030054 (1442) <br> cell junction <br> 2.635e-03","GO:0034703 (94) <br> cation channel complex <br> 1.202e-02","GO:0045202 (1073) <br> synapse <br> 1.410e-02","GO:0098797 (340) <br> plasma membrane protein complex <br> 3.668e-02","GO:0045211 (159) <br> postsynaptic membrane <br> 4.933e-02"],"key":["GO:0071944","GO:0005886","GO:0036477","GO:0005576","GO:0005615","GO:0043005","GO:0042995","GO:0120025","GO:0030424","GO:0098590","GO:0044297","GO:0043025","GO:0031012","GO:0030312","GO:0030425","GO:0097447","GO:0097060","GO:0062023","GO:0030054","GO:0034703","GO:0045202","GO:0098797","GO:0045211"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.80000000000000004,"size":[21.36096400395196,21.179422599094369,18.270279735351277,18.840197059751805,18.542360893284261,18.879680264336315,19.945614989254661,19.91584665478473,17.306276955306224,18.444562270240251,17.684990059450648,17.332351927824497,15.341296037944554,15.353824968653701,17.352985456048224,17.363227993376825,15.613839115981531,14.696411554466177,19.841532934725052,12.146718599567423,19.195091115242032,16.369972855763891,14.084622116091404],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData1f2373df","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[14.735789185520002,21.720992590609537,4.84619841750298,20.701421068606244,5.1801523651566628,19.019001559593374,11.388659846536497,11.391189800685387,5.220631631538927],"y":[6.3679154715236717,6.3679154715236717,4.0962907706029634,3.208738536497548,2.5892368078313988,2.0525641527918861,1.7462196439669426,1.7104717571550645,1.5577642575048973],"text":["GO:0038023 (497) <br> signaling receptor activity <br> 4.286e-07","GO:0060089 (497) <br> molecular transducer activity <br> 4.286e-07","GO:0004888 (371) <br> transmembrane signaling receptor activity <br> 8.011e-05","GO:0050839 (190) <br> cell adhesion molecule binding <br> 6.184e-04","GO:0005102 (935) <br> signaling receptor binding <br> 2.575e-03","GO:0048018 (173) <br> receptor ligand activity <br> 8.860e-03","GO:0030545 (198) <br> signaling receptor regulator activity <br> 1.794e-02","GO:0030546 (178) <br> signaling receptor activator activity <br> 1.948e-02","GO:0005125 (79) <br> cytokine activity <br> 2.768e-02"],"key":["GO:0038023","GO:0060089","GO:0004888","GO:0050839","GO:0005102","GO:0048018","GO:0030545","GO:0030546","GO:0005125"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.80000000000000004,"size":[17.36833090125727,17.36833090125727,16.606312072309283,14.66334441452851,18.884560808688352,14.362747521044881,14.793010303914265,14.45501568645078,11.398028524048092],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData1f2373df","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[113.09563933416906,113.12098084163175],"y":[8.278493169632279,2.2542968929461344],"text":["KEGG:04060 (183) <br> Cytokine-cytokine receptor interaction <br> 5.266e-09","KEGG:04080 (123) <br> Neuroactive ligand-receptor interaction <br> 5.568e-03"],"key":["KEGG:04060","KEGG:04080"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.80000000000000004,"size":[14.543969322012281,13.188105377635102],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData1f2373df","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[120.08361715866145,117.86888116566276,117.62083073444691],"y":[2.6185501024137525,2.6012785090433437,1.9246802565760548],"text":["REAC:R-MMU-372790 (325) <br> Signaling by GPCR <br> 2.407e-03","REAC:R-MMU-388396 (309) <br> GPCR downstream signalling <br> 2.505e-03","REAC:R-MMU-1474244 (144) <br> Extracellular matrix organization <br> 1.189e-02"],"key":["REAC:R-MMU-372790","REAC:R-MMU-388396","REAC:R-MMU-1474244"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(51,102,204,1)","opacity":0.80000000000000004,"size":[16.246004153863868,16.105821468110555,13.748135962630579],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(51,102,204,1)"}},"hoveron":"points","set":"SharedData1f2373df","name":"REAC","legendgroup":"REAC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[131.91496083604028,126.53555638946405,126.92775191590776,133.15480475834616,132.89924509272802],"y":[5.6443571256505711,2.5039362007303048,1.6902401559816187,1.537881114177877,1.3474809763331441],"text":["TF:M10376 (3559) <br> Factor: Sp5; motif: RNGGRGGNGGRGNNGGGGGAGGRG <br> 2.268e-06","TF:M01587 (1740) <br> Factor: FPM315; motif: SRGGGAGGAGGN <br> 3.134e-03","TF:M07040_1 (6086) <br> Factor: GKLF; motif: NNRRGRRNGNSNNN; match class: 1 <br> 2.041e-02","TF:M07436 (5624) <br> Factor: WT1; motif: NNGGGNGGGSGN <br> 2.898e-02","TF:M00706_1 (957) <br> Factor: TFII-I; motif: RGAGGKAGG; match class: 1 <br> 4.493e-02"],"key":["TF:M10376","TF:M01587","TF:M07040_1","TF:M07436","TF:M00706_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.80000000000000004,"size":[21.673243959701878,20.239195064287951,22.677165354330711,22.532810478690493,18.937466942498556],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedData1f2373df","name":"TF","legendgroup":"TF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,27.673974702951323],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[30.203679570716751,40.378152548869295],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[42.907857416634727,109.92732847834419],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[112.45703334610963,113.89896512073591],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[116.42866998850134,120.93154465312381],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(51,102,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[123.46124952088923,134.33139133767727],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[136.86109620544272,142.02422384055194],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[144.55392870831739,146.27159831353009],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[148.80130318129554,196.95170563434269],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[199.4814105021081,200.00252970486778],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(0,153,198,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.199999999999999],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.559055118110237,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.75590551181102372,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.283520132835207,"r":6.6417600664176026,"b":55.878166138820198,"l":61.037775010377771},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"tickvals":[14.836987351475663,35.29091605979302,76.417592947489453,113.17799923342277,118.68010732081255,128.89632042928326,139.44266002299733,145.41276351092375,172.87650440781911,199.74197010348794],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.62681610626816},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.13283520132835203,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.393109173931101},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.973017849730184,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"3743fa6dcbb682":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"3743fa74bb7144":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa348de97a":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa7c37b95e":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa6f6f6790":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa5f8c5296":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa500cbdf":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa36b9747a":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa3a57d871":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa44051f6f":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa46f35c87":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa1d535600":{"x":{},"y":{}},"3743fa4957e1cf":{"yintercept":{}}},"cur_data":"3743fa6dcbb682","visdat":{"3743fa6dcbb682":["function (y) ","x"],"3743fa74bb7144":["function (y) ","x"],"3743fa348de97a":["function (y) ","x"],"3743fa7c37b95e":["function (y) ","x"],"3743fa6f6f6790":["function (y) ","x"],"3743fa5f8c5296":["function (y) ","x"],"3743fa500cbdf":["function (y) ","x"],"3743fa36b9747a":["function (y) ","x"],"3743fa3a57d871":["function (y) ","x"],"3743fa44051f6f":["function (y) ","x"],"3743fa46f35c87":["function (y) ","x"],"3743fa1d535600":["function (y) ","x"],"3743fa4957e1cf":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData1f2373df"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">## size of circle - term size</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="3" type="a">
<li>What does <code>as_short_link = TRUE</code> do in <code>gost()</code> function?</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>gost_results_obj_link <span class="ot">&lt;-</span> <span class="fu">gost</span>(<span class="at">query =</span> upreg_gene_ids, </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">organism =</span> <span class="st">"mmusculus"</span>, </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ordered_query =</span> <span class="cn">FALSE</span>,    </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">multi_query =</span> <span class="cn">FALSE</span>, </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">significant =</span> <span class="cn">TRUE</span>, </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">exclude_iea =</span> <span class="cn">TRUE</span>, </span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">measure_underrepresentation =</span> <span class="cn">FALSE</span>, </span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">evcodes =</span> <span class="cn">FALSE</span>, </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">user_threshold =</span> <span class="fl">0.05</span>, </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">correction_method =</span> <span class="st">"g_SCS"</span>,  </span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">domain_scope =</span> <span class="st">"custom_annotated"</span>,</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">custom_bg =</span> all_gene_tested_ids, </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                         <span class="at">numeric_ns =</span> <span class="st">""</span>, </span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sources =</span> <span class="cn">NULL</span>, </span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">as_short_link =</span> <span class="cn">TRUE</span>,</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>                         <span class="at">highlight =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>gost_results_obj_link</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "https://biit.cs.ut.ee/gplink/l/aFHQvgjxoS-"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This gives you a url where you can quickly and easily visualise your data and send the link to others to see it too</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Check 'Detailed Results' tab; see g:Profiler docs (https://biit.cs.ut.ee/gprofiler/page/docs) to learn more about the website</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="4" type="a">
<li>Filter results table for gene sets with between 5-300 genes, inclusive. Examine results subset.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>gost_results_obj_filtered_df <span class="ot">&lt;-</span> gost_results_obj<span class="sc">$</span>result</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>gost_results_obj_filtered_df <span class="ot">&lt;-</span> gost_results_obj_filtered_df <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(term_size <span class="sc">&gt;=</span> <span class="dv">5</span> <span class="sc">&amp;</span> term_size <span class="sc">&lt;=</span> <span class="dv">300</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p_value)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>gost_results_obj_filtered <span class="ot">&lt;-</span> gost_results_obj</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>gost_results_obj_filtered<span class="sc">$</span>result <span class="ot">&lt;-</span> gost_results_obj_filtered_df</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="fu">gostplot</span>(gost_results_obj_filtered, <span class="at">capped =</span> <span class="cn">TRUE</span>, <span class="at">interactive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-d132595cb959830d9f40" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-d132595cb959830d9f40">{"x":{"data":[{"x":[43.714863730566684,43.836294147710042,93.802381001848019,69.417635359245836,76.096308302130964,50.256927454165528,90.144289685404118,86.005536301101074,65.5116236078009,43.83376434735289,71.593263666397803,59.872698611705822,49.417033735590579,68.446192022098899,69.490999569603275],"y":[7.79750401134031,4.2056291610934711,4.0107938070502982,3.9278551494360165,3.8831559131422444,2.2910795770962467,2.2910795770962467,1.6440999518520778,1.4124466377366311,1.403910306167045,1.3955945294669672,1.3955945294669672,1.3432031477291331,1.3432031477291331,1.3345771281489549],"text":["GO:0001503 (276) <br> ossification <br> 1.594e-08","GO:0001654 (259) <br> eye development <br> 6.228e-05","GO:0150063 (262) <br> visual system development <br> 9.755e-05","GO:0043010 (229) <br> camera-type eye development <br> 1.181e-04","GO:0048880 (264) <br> sensory system development <br> 1.309e-04","GO:0007411 (139) <br> axon guidance <br> 5.116e-03","GO:0097485 (139) <br> neuron projection guidance <br> 5.116e-03","GO:0072001 (202) <br> renal system development <br> 2.269e-02","GO:0035272 (34) <br> exocrine system development <br> 3.869e-02","GO:0001649 (144) <br> osteoblast differentiation <br> 3.945e-02","GO:0045229 (192) <br> external encapsulating structure organization <br> 4.022e-02","GO:0030198 (192) <br> extracellular matrix organization <br> 4.022e-02","GO:0006935 (288) <br> chemotaxis <br> 4.537e-02","GO:0042330 (288) <br> taxis <br> 4.537e-02","GO:0043062 (193) <br> extracellular structure organization <br> 4.628e-02"],"key":["GO:0001503","GO:0001654","GO:0150063","GO:0043010","GO:0048880","GO:0007411","GO:0097485","GO:0072001","GO:0035272","GO:0001649","GO:0045229","GO:0030198","GO:0006935","GO:0042330","GO:0043062"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.80000000000000004,"size":[22.488004597343814,22.201831296873188,22.2540014024101,21.634632906747349,22.28837003983487,19.120918258112461,19.120918258112461,21.03744755091228,3.7795275590551185,19.31223645758163,20.789803649234706,20.789803649234706,22.677165354330711,22.677165354330711,20.815306883934568],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData3c4966fc","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[32.741604563774949,32.488571164666453,37.389828105397982,36.033569086176456,34.021953563263928,35.532562955941636],"y":[4.3305697254076563,4.2610502226644318,3.715612311093202,2.7640036490245441,1.9201502931132151,1.3069244560821554],"text":["GO:0031012 (237) <br> extracellular matrix <br> 4.671e-05","GO:0030312 (238) <br> external encapsulating structure <br> 5.482e-05","GO:0097060 (260) <br> synaptic membrane <br> 1.925e-04","GO:0062023 (192) <br> collagen-containing extracellular matrix <br> 1.722e-03","GO:0034703 (94) <br> cation channel complex <br> 1.202e-02","GO:0045211 (159) <br> postsynaptic membrane <br> 4.933e-02"],"key":["GO:0031012","GO:0030312","GO:0097060","GO:0062023","GO:0034703","GO:0045211"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.80000000000000004,"size":[21.794638281062909,21.814160387422834,22.219304605796701,20.789803649234706,16.816968440374314,19.836536466017844],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData3c4966fc","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[20.701421068606244,19.019001559593374,11.388659846536497,11.391189800685387,5.220631631538927],"y":[3.208738536497548,2.0525641527918861,1.7462196439669426,1.7104717571550645,1.5577642575048973],"text":["GO:0050839 (190) <br> cell adhesion molecule binding <br> 6.184e-04","GO:0048018 (173) <br> receptor ligand activity <br> 8.860e-03","GO:0030545 (198) <br> signaling receptor regulator activity <br> 1.794e-02","GO:0030546 (178) <br> signaling receptor activator activity <br> 1.948e-02","GO:0005125 (79) <br> cytokine activity <br> 2.768e-02"],"key":["GO:0050839","GO:0048018","GO:0030545","GO:0030546","GO:0005125"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.80000000000000004,"size":[20.738279681398396,20.269900961595546,20.940320170136534,20.413669730096565,15.650387857864269],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData3c4966fc","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[113.09563933416906,113.12098084163175],"y":[8.278493169632279,2.2542968929461344],"text":["KEGG:04060 (183) <br> Cytokine-cytokine receptor interaction <br> 5.266e-09","KEGG:04080 (123) <br> Neuroactive ligand-receptor interaction <br> 5.568e-03"],"key":["KEGG:04060","KEGG:04080"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.80000000000000004,"size":[20.552273923450855,18.439617963721819],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedData3c4966fc","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[117.62083073444691],"y":[1.9246802565760548],"text":"REAC:R-MMU-1474244 (144) <br> Extracellular matrix organization <br> 1.189e-02","key":["REAC:R-MMU-1474244"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(51,102,204,1)","opacity":0.80000000000000004,"size":19.31223645758163,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(51,102,204,1)"}},"hoveron":"points","set":"SharedData3c4966fc","name":"REAC","legendgroup":"REAC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[2,27.673974702951323],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[30.203679570716751,40.378152548869295],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[42.907857416634727,109.92732847834419],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[112.45703334610963,113.89896512073591],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[116.42866998850134,120.93154465312381],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(51,102,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[123.46124952088923,134.33139133767727],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[136.86109620544272,142.02422384055194],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[144.55392870831739,146.27159831353009],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[148.80130318129554,196.95170563434269],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[199.4814105021081,200.00252970486778],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(0,153,198,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.199999999999999],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.559055118110237,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.75590551181102372,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.283520132835207,"r":6.6417600664176026,"b":55.878166138820198,"l":61.037775010377771},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"tickvals":[14.836987351475663,35.29091605979302,76.417592947489453,113.17799923342277,118.68010732081255,128.89632042928326,139.44266002299733,145.41276351092375,172.87650440781911,199.74197010348794],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.62681610626816},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.13283520132835203,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.393109173931101},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.973017849730184,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"3743fa2d2e95df":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"3743fa143be5e8":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa57c64dce":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa21ea0724":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa48c9cf62":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa53fe072d":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa11596eb4":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa285621f9":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa58fed30c":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa4812e32e":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa62adfa6a":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa1d03f27b":{"x":{},"y":{}},"3743faf063fb6":{"yintercept":{}}},"cur_data":"3743fa2d2e95df","visdat":{"3743fa2d2e95df":["function (y) ","x"],"3743fa143be5e8":["function (y) ","x"],"3743fa57c64dce":["function (y) ","x"],"3743fa21ea0724":["function (y) ","x"],"3743fa48c9cf62":["function (y) ","x"],"3743fa53fe072d":["function (y) ","x"],"3743fa11596eb4":["function (y) ","x"],"3743fa285621f9":["function (y) ","x"],"3743fa58fed30c":["function (y) ","x"],"3743fa4812e32e":["function (y) ","x"],"3743fa62adfa6a":["function (y) ","x"],"3743fa1d03f27b":["function (y) ","x"],"3743faf063fb6":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData3c4966fc"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<ol start="5" type="a">
<li>Try querying multiple gene lists at the same time (up and down regulated genes) using the parameter to specify multiple queries.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>multi_gostquery_results_obj <span class="ot">&lt;-</span> <span class="fu">gost</span>(<span class="at">query =</span> <span class="fu">list</span>(<span class="st">"upreg_genes"</span> <span class="ot">=</span> upreg_gene_ids,</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">"downreg_genes"</span><span class="ot">=</span> downreg_gene_ids),</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">organism =</span> <span class="st">"mmusculus"</span>, </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">multi_query =</span> <span class="cn">FALSE</span>,</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">exclude_iea =</span> <span class="cn">TRUE</span>, </span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">correction_method =</span> <span class="st">"g_SCS"</span>, </span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">domain_scope =</span> <span class="st">"custom_annotated"</span>,</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">custom_bg =</span> all_gene_tested_ids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="6" type="a">
<li>Can you plot these queries together?</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gostplot</span>(multi_gostquery_results_obj, <span class="at">capped =</span> <span class="cn">TRUE</span>, <span class="at">interactive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-d707b9608c646ea2ccd9" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-d707b9608c646ea2ccd9">{"x":{"data":[{"x":[49.447391339876425,45.19732673985861,45.349114761287815,76.650334580347575,44.888691096285882,45.364293563430735,44.883631495571578,58.132195965984252,55.794660435974457,45.392121367359422,51.84311227810074,75.425911207485299,49.439801938804962,69.660496193532566,78.079671782139272,45.946147645576033,51.848171878815052,49.809152790949362,90.450395528619694,45.941088044861729,49.811682591306521,71.633740472112265,93.045970695059154,44.018439773425087,76.430241949275228,70.728071944251326,71.41617764139707,49.781324987020675,76.425182348560909,59.703201987776538,51.868410281672283,56.133653683833018,77.282784669635944,75.423381407128147,62.159638134572553,61.537307246712807,76.078599699630885,61.542366847427111,49.783854787377827,73.98645480426498,67.947821351739663,46.130823071648237,44.891220896643034,46.113114469148158,45.979035050219025,60.262287866707467,79.683565208574578,44.230943003425978,45.994213852361945,46.140942273076845,77.285314469993082,60.968102166353269,45.313697556287664,49.417033735590579,68.446192022098899,49.444861539519266,76.657923981419032,60.963042565638965,76.554202166775738,72.25354155961486,69.05081430745858,44.820386486642739,106.95481305868887,106.94216405690311,49.847099796306665,44.236002604140296,97.045585059718761,60.947863763496045,45.984094650933329,78.092320783925047,44.944346704143264,59.791745000276919,45.971445649147562,59.629837777419098,77.300493272136009,76.584559771061592,46.302849495934666,90.184766491118594,46.297789895220362,59.82463240491991,45.543909388788634,60.952923364210356,44.238532404497441,46.024571456647791,49.885046801663961,75.28424238748471,90.478223332548396,43.740161734138212,85.524874233241917,75.289301988199014,68.372827811741445,71.418707441754222,59.822102604562758,76.076069899273733,53.765760549537369,71.421237242111374,43.798347142352746,72.294018365329322,59.705731788133697,60.259758066350308,75.208348376770104,60.057374037778025,90.187296291475747,83.999404617878383,52.531217975246484,67.950351152096829,46.029631057362096,49.793973988806449,45.999273453076249,44.309366814497736,45.351644561644967,107.39752812119073,44.853273891285731,44.830505688071355,52.528688174889339,106.47415099082971,85.408503416812849,44.101923185211163,58.129666165627093,68.367768211027141,72.620362611402101,51.939244691672577,65.473676602443589,53.763230749180224,93.57975857041852,59.602009973490411,66.948550210664052,59.700672187419393,76.308811532131855,83.994345017164065,62.149518933143938,54.299548424896756,59.796804600991223,76.549142566061434,68.069251768883035,62.964114648147358,44.104452985568308,69.738920004604324,72.256071359972012,44.243592005211752,73.988984604622132,83.154451298589123,46.11564426950531,76.453010152489611,59.596950372776107,44.946876504500409,102.25191419474059,46.436928914863799,68.327291405312693,64.608484880297112,62.961584847790206],"y":[17,17,17,17,17,17,17,17,17,17,14.63351825836652,12.90042156220829,12.465052915147137,12.384277173459722,12.384277173459722,12.037713077165821,11.766436192963612,11.647021344685207,11.021596226029226,11.018109822466245,10.997237953820505,10.858548161405212,10.784598751118569,10.499444375484336,10.109844677197652,9.753164167409679,9.7139318214992159,9.5068224971492281,9.3901765394006667,9.132461682085717,8.9217803912796683,8.5194595118621415,8.5044509833310755,8.4804383848648008,7.5604976698278268,7.4316085013645345,7.391278047850169,7.3912037253910459,7.2451917223957842,7.2445708872353434,6.791821391857872,6.7707364615656518,6.7054604623109872,6.6218028310031434,6.4634480164467698,6.4556652077706547,6.3007829012969871,6.1736101249523703,6.0081581271312903,5.991666843535997,5.966148902167915,5.91072527821375,5.8554053000966872,5.7410725291283962,5.7410725291283962,5.5760368422667517,5.3941808594030345,5.3827911426453161,5.3602991150321362,5.1141563429819801,5.0890918998542123,5.0280615998109974,5.0280615998109974,5.0280615998109974,5.0219855865461289,4.8663050391977549,4.769554130601839,4.6141920775152938,4.5873694160462231,4.4519987113606874,4.3740246382146513,4.3469172037241552,4.163464901842163,4.0639463114931083,4.0532202195767546,3.8927085677885875,3.7924101783031046,3.7103940862952003,3.6614527509426953,3.6315550214319265,3.5131998734911072,3.508669988115301,3.4960525697409932,3.4770084770959397,3.4489370058939062,3.4468389319833719,3.4187555697999681,3.3747518711249276,3.2343841108312286,3.2117124761695535,3.0789956793573898,3.0535609566606117,3.0312920356054978,3.0312920356054978,2.9735998796861369,2.971221933225054,2.9529750115434474,2.9290079008133416,2.9093416794385414,2.8411980134035963,2.7578278821060103,2.7364141607355461,2.7111772665979283,2.6605850387598471,2.648173704258121,2.6010872390735207,2.5960716336665066,2.5821888404399918,2.5573119301598184,2.5279688646534311,2.5040626659353502,2.4870145573437252,2.4764911927346338,2.4439486454222812,2.4367726270186951,2.2224400616495812,2.2224400616495812,2.1995422065160648,2.0819154264216309,2.0587357844627743,2.0446703564142066,2.025177273516352,2.0089003388289606,2.006919637057635,1.9057360616559502,1.8949849509159535,1.8922072868473025,1.8732563068063757,1.8653465893240351,1.8401759395560611,1.8378521388664151,1.8377626058793322,1.8181760501446829,1.8156671194964693,1.787758782611389,1.7068348502245754,1.7065630366751989,1.6602609410066784,1.6066564322964381,1.6032772517757072,1.5675952319623878,1.5542456282971544,1.4970781347339623,1.4831609964927619,1.4808994221904905,1.4738881229796805,1.4647451461185885,1.4326788806793265,1.4105517955827871,1.4049704515947194,1.3471816068319971],"text":["GO:0006955 (1220) <br> immune response <br> 3.187e-77","GO:0002376 (1902) <br> immune system process <br> 3.957e-73","GO:0002443 (370) <br> leukocyte mediated immunity <br> 8.676e-41","GO:0050896 (5387) <br> response to stimulus <br> 2.470e-39","GO:0002252 (552) <br> immune effector process <br> 3.893e-36","GO:0002449 (295) <br> lymphocyte mediated immunity <br> 1.282e-33","GO:0002250 (339) <br> adaptive immune response <br> 2.345e-30","GO:0019724 (169) <br> B cell mediated immunity <br> 7.475e-28","GO:0016064 (165) <br> immunoglobulin mediated immune response <br> 1.220e-27","GO:0002460 (313) <br> adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains <br> 1.461e-26","GO:0009605 (1529) <br> response to external stimulus <br> 2.325e-15","GO:0048584 (1713) <br> positive regulation of response to stimulus <br> 1.258e-13","GO:0006952 (1131) <br> defense response <br> 3.427e-13","GO:0043207 (1012) <br> response to external biotic stimulus <br> 4.128e-13","GO:0051707 (1012) <br> response to other organism <br> 4.128e-13","GO:0002684 (863) <br> positive regulation of immune system process <br> 9.168e-13","GO:0009607 (1038) <br> response to biotic stimulus <br> 1.712e-12","GO:0007165 (3410) <br> signal transduction <br> 2.254e-12","GO:0098542 (771) <br> defense response to other organism <br> 9.515e-12","GO:0002682 (1213) <br> regulation of immune system process <br> 9.592e-12","GO:0007166 (1812) <br> cell surface receptor signaling pathway <br> 1.006e-11","GO:0045321 (794) <br> leukocyte activation <br> 1.385e-11","GO:0140546 (691) <br> defense response to symbiont <br> 1.642e-11","GO:0001775 (871) <br> cell activation <br> 3.166e-11","GO:0050778 (594) <br> positive regulation of immune response <br> 7.765e-11","GO:0044419 (1125) <br> biological process involved in interspecies interaction between organisms <br> 1.765e-10","GO:0045087 (593) <br> innate immune response <br> 1.932e-10","GO:0007154 (3732) <br> cell communication <br> 3.113e-10","GO:0050776 (713) <br> regulation of immune response <br> 4.072e-10","GO:0023052 (3726) <br> signaling <br> 7.371e-10","GO:0009617 (491) <br> response to bacterium <br> 1.197e-09","GO:0016477 (1089) <br> cell migration <br> 3.024e-09","GO:0051239 (2176) <br> regulation of multicellular organismal process <br> 3.130e-09","GO:0048583 (2828) <br> regulation of response to stimulus <br> 3.308e-09","GO:0032501 (4443) <br> multicellular organismal process <br> 2.751e-08","GO:0032101 (812) <br> regulation of response to external stimulus <br> 3.702e-08","GO:0048870 (1200) <br> cell motility <br> 4.062e-08","GO:0032103 (513) <br> positive regulation of response to external stimulus <br> 4.063e-08","GO:0007155 (868) <br> cell adhesion <br> 5.686e-08","GO:0046649 (678) <br> lymphocyte activation <br> 5.694e-08","GO:0040011 (871) <br> locomotion <br> 1.615e-07","GO:0002764 (399) <br> immune response-regulating signaling pathway <br> 1.695e-07","GO:0002253 (436) <br> activation of immune response <br> 1.970e-07","GO:0002757 (393) <br> immune response-activating signaling pathway <br> 2.389e-07","GO:0002697 (335) <br> regulation of immune effector process <br> 3.440e-07","GO:0030595 (161) <br> leukocyte chemotaxis <br> 3.502e-07","GO:0060326 (215) <br> cell chemotaxis <br> 5.003e-07","GO:0001906 (157) <br> cell killing <br> 6.705e-07","GO:0002703 (219) <br> regulation of leukocyte mediated immunity <br> 9.814e-07","GO:0002768 (243) <br> immune response-regulating cell surface receptor signaling pathway <br> 1.019e-06","GO:0051240 (1284) <br> positive regulation of multicellular organismal process <br> 1.081e-06","GO:0031349 (416) <br> positive regulation of defense response <br> 1.228e-06","GO:0002429 (237) <br> immune response-activating cell surface receptor signaling pathway <br> 1.395e-06","GO:0006935 (288) <br> chemotaxis <br> 1.815e-06","GO:0042330 (288) <br> taxis <br> 1.815e-06","GO:0006954 (545) <br> inflammatory response <br> 2.654e-06","GO:0050900 (277) <br> leukocyte migration <br> 4.035e-06","GO:0031347 (628) <br> regulation of defense response <br> 4.142e-06","GO:0050853 (51) <br> B cell receptor signaling pathway <br> 4.362e-06","GO:0045765 (208) <br> regulation of angiogenesis <br> 7.689e-06","GO:0042742 (185) <br> defense response to bacterium <br> 8.145e-06","GO:0002223 (12) <br> stimulatory C-type lectin receptor signaling pathway <br> 9.374e-06","GO:1990858 (12) <br> cellular response to lectin <br> 9.374e-06","GO:1990840 (12) <br> response to lectin <br> 9.374e-06","GO:0007186 (389) <br> G protein-coupled receptor signaling pathway <br> 9.506e-06","GO:0001909 (115) <br> leukocyte mediated cytotoxicity <br> 1.360e-05","GO:1901342 (213) <br> regulation of vasculature development <br> 1.700e-05","GO:0031341 (104) <br> regulation of cell killing <br> 2.431e-05","GO:0002699 (232) <br> positive regulation of immune effector process <br> 2.586e-05","GO:0051716 (4337) <br> cellular response to stimulus <br> 3.532e-05","GO:0002274 (187) <br> myeloid leukocyte activation <br> 4.226e-05","GO:0030097 (812) <br> hemopoiesis <br> 4.499e-05","GO:0002694 (474) <br> regulation of leukocyte activation <br> 6.863e-05","GO:0022603 (645) <br> regulation of anatomical structure morphogenesis <br> 8.631e-05","GO:0051249 (429) <br> regulation of lymphocyte activation <br> 8.847e-05","GO:0050865 (510) <br> regulation of cell activation <br> 1.280e-04","GO:0002833 (342) <br> positive regulation of response to biotic stimulus <br> 1.613e-04","GO:0097529 (165) <br> myeloid leukocyte migration <br> 1.948e-04","GO:0002831 (448) <br> regulation of response to biotic stimulus <br> 2.180e-04","GO:0030155 (607) <br> regulation of cell adhesion <br> 2.336e-04","GO:0002521 (550) <br> leukocyte differentiation <br> 3.068e-04","GO:0031343 (75) <br> positive regulation of cell killing <br> 3.100e-04","GO:0001910 (88) <br> regulation of leukocyte mediated cytotoxicity <br> 3.191e-04","GO:0002715 (63) <br> regulation of natural killer cell mediated immunity <br> 3.334e-04","GO:0007204 (116) <br> positive regulation of cytosolic calcium ion concentration <br> 3.557e-04","GO:0048514 (415) <br> blood vessel morphogenesis <br> 3.574e-04","GO:0098609 (593) <br> cell-cell adhesion <br> 3.813e-04","GO:0001525 (351) <br> angiogenesis <br> 4.219e-04","GO:0071674 (156) <br> mononuclear cell migration <br> 5.829e-04","GO:0048518 (4454) <br> positive regulation of biological process <br> 6.142e-04","GO:0042269 (60) <br> regulation of natural killer cell mediated cytotoxicity <br> 8.337e-04","GO:0045088 (377) <br> regulation of innate immune response <br> 8.840e-04","GO:0030154 (2809) <br> cell differentiation <br> 9.305e-04","GO:0048869 (2809) <br> cellular developmental process <br> 9.305e-04","GO:0010647 (1319) <br> positive regulation of cell communication <br> 1.063e-03","GO:0045089 (323) <br> positive regulation of innate immune response <br> 1.069e-03","GO:0001568 (486) <br> blood vessel development <br> 1.114e-03","GO:0045785 (399) <br> positive regulation of cell adhesion <br> 1.178e-03","GO:0023056 (1322) <br> positive regulation of signaling <br> 1.232e-03","GO:0030593 (68) <br> neutrophil chemotaxis <br> 1.441e-03","GO:0048468 (2116) <br> cell development <br> 1.747e-03","GO:0030334 (708) <br> regulation of cell migration <br> 1.835e-03","GO:0097530 (110) <br> granulocyte migration <br> 1.945e-03","GO:0070663 (231) <br> regulation of leukocyte proliferation <br> 2.185e-03","GO:0009967 (1151) <br> positive regulation of signal transduction <br> 2.248e-03","GO:0040012 (766) <br> regulation of locomotion <br> 2.506e-03","GO:0002717 (46) <br> positive regulation of natural killer cell mediated immunity <br> 2.535e-03","GO:0007159 (341) <br> leukocyte cell-cell adhesion <br> 2.617e-03","GO:0002705 (142) <br> positive regulation of leukocyte mediated immunity <br> 2.771e-03","GO:0001944 (508) <br> vasculature development <br> 2.965e-03","GO:0002444 (91) <br> myeloid leukocyte mediated immunity <br> 3.133e-03","GO:2000145 (738) <br> regulation of cell motility <br> 3.258e-03","GO:0002237 (252) <br> response to molecule of bacterial origin <br> 3.338e-03","GO:0002228 (78) <br> natural killer cell mediated immunity <br> 3.598e-03","GO:0009966 (2123) <br> regulation of signal transduction <br> 3.658e-03","GO:1990266 (87) <br> neutrophil migration <br> 5.992e-03","GO:0071621 (87) <br> granulocyte chemotaxis <br> 5.992e-03","GO:0001816 (642) <br> cytokine production <br> 6.316e-03","GO:0019722 (133) <br> calcium-mediated signaling <br> 8.281e-03","GO:0042267 (75) <br> natural killer cell mediated cytotoxicity <br> 8.735e-03","GO:0045954 (44) <br> positive regulation of natural killer cell mediated cytotoxicity <br> 9.023e-03","GO:0009653 (1806) <br> anatomical structure morphogenesis <br> 9.437e-03","GO:0035239 (596) <br> tube morphogenesis <br> 9.797e-03","GO:0010646 (2436) <br> regulation of cell communication <br> 9.842e-03","GO:0141124 (1220) <br> intracellular signaling cassette <br> 1.242e-02","GO:0022409 (273) <br> positive regulation of cell-cell adhesion <br> 1.274e-02","GO:0036230 (34) <br> granulocyte activation <br> 1.282e-02","GO:0023051 (2445) <br> regulation of signaling <br> 1.339e-02","GO:0050670 (211) <br> regulation of lymphocyte proliferation <br> 1.363e-02","GO:0070661 (302) <br> leukocyte proliferation <br> 1.445e-02","GO:0032496 (238) <br> response to lipopolysaccharide <br> 1.453e-02","GO:0010919 (9) <br> regulation of inositol phosphate biosynthetic process <br> 1.453e-02","GO:0030099 (379) <br> myeloid cell differentiation <br> 1.520e-02","GO:0050851 (161) <br> antigen receptor-mediated signaling pathway <br> 1.529e-02","GO:0042060 (248) <br> wound healing <br> 1.630e-02","GO:0032944 (214) <br> regulation of mononuclear cell proliferation <br> 1.964e-02","GO:0001817 (638) <br> regulation of cytokine production <br> 1.965e-02","GO:0043299 (59) <br> leukocyte degranulation <br> 2.186e-02","GO:0045766 (124) <br> positive regulation of angiogenesis <br> 2.474e-02","GO:0001912 (66) <br> positive regulation of leukocyte mediated cytotoxicity <br> 2.493e-02","GO:0046651 (271) <br> lymphocyte proliferation <br> 2.706e-02","GO:0070098 (36) <br> chemokine-mediated signaling pathway <br> 2.791e-02","GO:0002758 (227) <br> innate immune response-activating signaling pathway <br> 3.184e-02","GO:0050793 (1802) <br> regulation of developmental process <br> 3.287e-02","GO:0022407 (398) <br> regulation of cell-cell adhesion <br> 3.304e-02","GO:0002275 (74) <br> myeloid cell activation involved in immune response <br> 3.358e-02","GO:1904018 (126) <br> positive regulation of vasculature development <br> 3.430e-02","GO:0002886 (61) <br> regulation of myeloid leukocyte mediated immunity <br> 3.693e-02","GO:0042221 (2276) <br> response to chemical <br> 3.886e-02","GO:0034341 (119) <br> response to type II interferon <br> 3.936e-02","GO:0032943 (276) <br> mononuclear cell proliferation <br> 4.496e-02"],"key":["GO:0006955","GO:0002376","GO:0002443","GO:0050896","GO:0002252","GO:0002449","GO:0002250","GO:0019724","GO:0016064","GO:0002460","GO:0009605","GO:0048584","GO:0006952","GO:0043207","GO:0051707","GO:0002684","GO:0009607","GO:0007165","GO:0098542","GO:0002682","GO:0007166","GO:0045321","GO:0140546","GO:0001775","GO:0050778","GO:0044419","GO:0045087","GO:0007154","GO:0050776","GO:0023052","GO:0009617","GO:0016477","GO:0051239","GO:0048583","GO:0032501","GO:0032101","GO:0048870","GO:0032103","GO:0007155","GO:0046649","GO:0040011","GO:0002764","GO:0002253","GO:0002757","GO:0002697","GO:0030595","GO:0060326","GO:0001906","GO:0002703","GO:0002768","GO:0051240","GO:0031349","GO:0002429","GO:0006935","GO:0042330","GO:0006954","GO:0050900","GO:0031347","GO:0050853","GO:0045765","GO:0042742","GO:0002223","GO:1990858","GO:1990840","GO:0007186","GO:0001909","GO:1901342","GO:0031341","GO:0002699","GO:0051716","GO:0002274","GO:0030097","GO:0002694","GO:0022603","GO:0051249","GO:0050865","GO:0002833","GO:0097529","GO:0002831","GO:0030155","GO:0002521","GO:0031343","GO:0001910","GO:0002715","GO:0007204","GO:0048514","GO:0098609","GO:0001525","GO:0071674","GO:0048518","GO:0042269","GO:0045088","GO:0030154","GO:0048869","GO:0010647","GO:0045089","GO:0001568","GO:0045785","GO:0023056","GO:0030593","GO:0048468","GO:0030334","GO:0097530","GO:0070663","GO:0009967","GO:0040012","GO:0002717","GO:0007159","GO:0002705","GO:0001944","GO:0002444","GO:2000145","GO:0002237","GO:0002228","GO:0009966","GO:1990266","GO:0071621","GO:0001816","GO:0019722","GO:0042267","GO:0045954","GO:0009653","GO:0035239","GO:0010646","GO:0141124","GO:0022409","GO:0036230","GO:0023051","GO:0050670","GO:0070661","GO:0032496","GO:0010919","GO:0030099","GO:0050851","GO:0042060","GO:0032944","GO:0001817","GO:0043299","GO:0045766","GO:0001912","GO:0046651","GO:0070098","GO:0002758","GO:0050793","GO:0022407","GO:0002275","GO:1904018","GO:0002886","GO:0042221","GO:0034341","GO:0032943"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.80000000000000004,"size":[20.338657165979015,21.016539043972376,18.362048817887739,22.509886295106604,19.053333261190048,17.955664566874034,18.206428020159283,16.90286692756732,16.855698066660825,18.063034001938977,20.686697419726901,20.859191948876994,20.220231048537165,20.044859904798905,20.044859904798905,19.790273506634897,20.085041576177257,21.869149810978495,19.607609694537867,20.329690696135167,20.943840662476873,19.655444606913857,19.428052425302422,19.805133382726073,19.176685227922441,20.211882954588635,19.173862084678287,21.997367787072275,19.479633076020058,21.995089316068526,18.854257697126044,20.160747292972655,21.216785622972814,21.600283231314172,22.242643247049507,19.691827166771731,20.312887467074798,18.9290775400979,19.799578598519425,19.396712429530904,19.805133382726073,18.494927441893402,18.649572448030572,18.468342295266687,18.185188914356992,16.807194452757518,17.367842883801224,16.757282318070146,17.402792552374621,17.598297762514498,20.418116747068208,18.567888512558845,17.551547639234933,17.91189747818218,17.91189747818218,19.031763711507178,17.840634371714135,19.269657684820931,14.287180880068256,17.304859840633391,17.079486953741071,9.7595782581721569,9.7595782581721569,9.7595782581721569,18.450365164236352,16.12297735388854,17.350089062310982,15.911040529954224,17.5115502773589,22.208874957271156,17.100326347774136,19.691827166771731,18.79383884705252,19.314075104437205,18.621468068166987,18.91908732995498,18.222173331804541,16.855698066660825,18.696598819151095,19.212913730864077,19.047201671735859,15.194778953223876,15.550462729336196,14.793731340751455,16.141058272199061,18.563689713679821,19.173862084678287,18.268494054153479,16.74457498180255,22.246097769095698,14.678862022317137,18.395165661924946,21.590522455953501,21.590522455953501,20.459760003583263,18.119718878619221,18.836732356095869,18.494927441893402,20.463273095242137,14.971173415121687,21.175372122803719,19.468066110795345,16.029727044491292,17.50343323490678,20.247711669364381,19.597002532832047,14.030735967020609,18.216942198637589,16.556119115595283,18.912390799325394,15.62370018126896,19.536154907814417,17.666020913129582,15.283068244864841,21.180268925995414,15.52538993348244,15.52538993348244,19.306331602497412,16.423227161498637,15.194778953223876,13.918260636661502,20.938855216613359,19.182315738216474,21.38296849771098,20.338657165979015,17.813923847453371,13.239545032802289,21.3883717376169,17.33214416196596,17.998275178774612,17.55943205209639,8.3532402425319212,18.404501149953479,16.807194452757518,17.636266152148171,17.358989614985362,19.295944392136693,14.639010745948134,16.279455481779483,14.902151176257419,17.800402098958742,13.394144578937755,17.470560749549829,20.935521567907546,18.490527782690883,15.164405287263902,16.312431903569362,14.717912962433546,21.283129153583115,16.19422697037108,17.833997818106482],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[62.159638134572553,75.762374654986715,75.281712587127558,49.993828217021566,76.045712294987894,62.162167934929712,51.939244691672577,52.387019354888743,49.781324987020675,77.282784669635944,59.703201987776538,76.076069899273733,59.822102604562758,77.285314469993082,43.714863730566684,76.078599699630885,52.384489554531584,46.596306337364467,50.236689051308304,49.783854787377827,49.811682591306521,59.738619192776689,50.287285058451367,92.16560017076975,75.208348376770104,76.453010152489611,49.976119614521494,56.133653683833018,67.947821351739663,49.809152790949362,77.029804633920591,67.950351152096829,107.13442888404676,91.262461443265963,71.861422504256069,43.836294147710042,59.505877559918574,75.681421043557805,93.802381001848019,67.963000153882604,79.908717440361244,91.264991243623115,69.417635359245836,59.852460208848598,76.096308302130964,68.155264981026249,49.978649414878646,90.842514583978485,75.549871424985824,90.478223332548396,50.879258342025274,71.866482104970387,75.600467432128895,64.590776277797033,65.473676602443589,65.5546302138725,43.512479701994401,107.39752812119073,60.057374037778025,60.831492947066991,82.271550973942567,76.607327974275975,50.254397653808368,86.847959820033168,50.874198741310977,107.40258772190502,75.602997232486047,77.027274833563439,50.256927454165528,90.144289685404118,68.327291405312693,76.050771895702212,75.939460679987448,92.168129971126902,83.106385091803205,76.316400933203326,75.757315054272411,60.059903838135185,60.457082494208279,86.005536301101074,78.092320783925047,53.899839968466502,43.709804129852373,65.5116236078009,43.83376434735289,71.593263666397803,59.872698611705822,77.287844270350249,49.417033735590579,68.446192022098899,69.490999569603275],"y":[17,15.336621779728503,13.412952046810675,12.376376781917749,11.702138346088722,10.774388309996146,9.9598249309555378,8.7229924852176612,8.5442148703601308,8.4708957775947926,8.4600019449909976,8.4062433167050479,8.4062433167050479,8.2060502989954269,7.79750401134031,7.5805604905910382,7.5370060042262548,7.4249302687416918,7.4168327212591354,7.2237359864308814,7.1934341482619919,6.1235988831261858,5.852314192075923,5.6357553535827316,5.5770847678200042,5.5519417686855856,5.4226676489692727,5.3270134990356297,5.2588912600100821,4.6743259489186295,4.6637035908267892,4.5654861363717805,4.5464879535712432,4.3405855555914412,4.2831779425815517,4.2056291610934711,4.0644954836548006,4.0383702326433717,4.0107938070502982,4.0040740479245835,3.9850568459474669,3.9461967490767362,3.9278551494360165,3.8948927167064613,3.8831559131422444,3.8598701710518024,3.8223546674174429,3.8223546674174429,3.7778063145907663,3.7136544367639575,3.6546530474573484,3.6546513421575018,3.6079214646647983,3.595352214746192,3.2837263637954899,3.2149206180457344,3.0887510652735473,2.9342228282757632,2.815784929174233,2.812153213543807,2.8081150751071515,2.7524836833077519,2.6644760776572634,2.4973806247212651,2.4160313313829733,2.4138134602322303,2.4082441900076419,2.3400208597968319,2.2910795770962467,2.2910795770962467,2.278025045600542,2.2766677503752799,2.2756089815593485,2.1584674132430255,2.1220899573934116,2.0154188406736386,1.8485581981036345,1.845186540655015,1.6879580841335329,1.6440999518520778,1.6296747832110996,1.5281985568046346,1.4736727744379268,1.4124466377366311,1.403910306167045,1.3955945294669672,1.3955945294669672,1.365647275498338,1.3432031477291331,1.3432031477291331,1.3345771281489549],"text":["GO:0032501 (4443) <br> multicellular organismal process <br> 6.352e-18","GO:0048731 (2609) <br> system development <br> 4.607e-16","GO:0048513 (1911) <br> animal organ development <br> 3.864e-14","GO:0007275 (3109) <br> multicellular organism development <br> 4.204e-13","GO:0048856 (3850) <br> anatomical structure development <br> 1.985e-12","GO:0032502 (4121) <br> developmental process <br> 1.681e-11","GO:0009653 (1806) <br> anatomical structure morphogenesis <br> 1.097e-10","GO:0009888 (1303) <br> tissue development <br> 1.892e-09","GO:0007154 (3732) <br> cell communication <br> 2.856e-09","GO:0051239 (2176) <br> regulation of multicellular organismal process <br> 3.381e-09","GO:0023052 (3726) <br> signaling <br> 3.467e-09","GO:0048869 (2809) <br> cellular developmental process <br> 3.924e-09","GO:0030154 (2809) <br> cell differentiation <br> 3.924e-09","GO:0051240 (1284) <br> positive regulation of multicellular organismal process <br> 6.222e-09","GO:0001503 (276) <br> ossification <br> 1.594e-08","GO:0048870 (1200) <br> cell motility <br> 2.627e-08","GO:0009887 (662) <br> animal organ morphogenesis <br> 2.904e-08","GO:0003008 (1086) <br> system process <br> 3.759e-08","GO:0007399 (1642) <br> nervous system development <br> 3.830e-08","GO:0007155 (868) <br> cell adhesion <br> 5.974e-08","GO:0007166 (1812) <br> cell surface receptor signaling pathway <br> 6.406e-08","GO:0030030 (1193) <br> cell projection organization <br> 7.523e-07","GO:0007423 (373) <br> sensory organ development <br> 1.405e-06","GO:0120036 (1174) <br> plasma membrane bounded cell projection organization <br> 2.313e-06","GO:0048468 (2116) <br> cell development <br> 2.648e-06","GO:0050793 (1802) <br> regulation of developmental process <br> 2.806e-06","GO:0007267 (784) <br> cell-cell signaling <br> 3.779e-06","GO:0016477 (1089) <br> cell migration <br> 4.710e-06","GO:0040011 (871) <br> locomotion <br> 5.509e-06","GO:0007165 (3410) <br> signal transduction <br> 2.117e-05","GO:0051094 (1007) <br> positive regulation of developmental process <br> 2.169e-05","GO:0040012 (766) <br> regulation of locomotion <br> 2.720e-05","GO:2000026 (1071) <br> regulation of multicellular organismal development <br> 2.841e-05","GO:0099536 (507) <br> synaptic signaling <br> 4.565e-05","GO:0045595 (1167) <br> regulation of cell differentiation <br> 5.210e-05","GO:0001654 (259) <br> eye development <br> 6.228e-05","GO:0022008 (1272) <br> neurogenesis <br> 8.620e-05","GO:0048699 (1088) <br> generation of neurons <br> 9.154e-05","GO:0150063 (262) <br> visual system development <br> 9.755e-05","GO:0040017 (445) <br> positive regulation of locomotion <br> 9.907e-05","GO:0060429 (774) <br> epithelium development <br> 1.035e-04","GO:0099537 (477) <br> trans-synaptic signaling <br> 1.132e-04","GO:0043010 (229) <br> camera-type eye development <br> 1.181e-04","GO:0030182 (1032) <br> neuron differentiation <br> 1.274e-04","GO:0048880 (264) <br> sensory system development <br> 1.309e-04","GO:0042127 (1182) <br> regulation of cell population proliferation <br> 1.381e-04","GO:0007268 (472) <br> chemical synaptic transmission <br> 1.505e-04","GO:0098916 (472) <br> anterograde trans-synaptic signaling <br> 1.505e-04","GO:0048646 (780) <br> anatomical structure formation involved in morphogenesis <br> 1.668e-04","GO:0098609 (593) <br> cell-cell adhesion <br> 1.934e-04","GO:0008285 (507) <br> negative regulation of cell population proliferation <br> 2.215e-04","GO:0045597 (684) <br> positive regulation of cell differentiation <br> 2.215e-04","GO:0048666 (878) <br> neuron development <br> 2.466e-04","GO:0034330 (540) <br> cell junction organization <br> 2.539e-04","GO:0035239 (596) <br> tube morphogenesis <br> 5.203e-04","GO:0035295 (738) <br> tube development <br> 6.096e-04","GO:0000902 (725) <br> cell morphogenesis <br> 8.152e-04","GO:2000145 (738) <br> regulation of cell motility <br> 1.164e-03","GO:0030334 (708) <br> regulation of cell migration <br> 1.528e-03","GO:0031175 (784) <br> neuron projection development <br> 1.541e-03","GO:0061564 (356) <br> axon development <br> 1.556e-03","GO:0050877 (635) <br> nervous system process <br> 1.768e-03","GO:0007409 (314) <br> axonogenesis <br> 2.165e-03","GO:0072359 (777) <br> circulatory system development <br> 3.181e-03","GO:0008283 (1438) <br> cell population proliferation <br> 3.837e-03","GO:2000147 (434) <br> positive regulation of cell motility <br> 3.856e-03","GO:0048667 (442) <br> cell morphogenesis involved in neuron differentiation <br> 3.906e-03","GO:0051093 (672) <br> negative regulation of developmental process <br> 4.571e-03","GO:0007411 (139) <br> axon guidance <br> 5.116e-03","GO:0097485 (139) <br> neuron projection guidance <br> 5.116e-03","GO:0042221 (2276) <br> response to chemical <br> 5.272e-03","GO:0048858 (493) <br> cell projection morphogenesis <br> 5.288e-03","GO:0048812 (477) <br> neuron projection morphogenesis <br> 5.301e-03","GO:0120039 (488) <br> plasma membrane bounded cell projection morphogenesis <br> 6.943e-03","GO:0065008 (1939) <br> regulation of biological quality <br> 7.549e-03","GO:0050673 (327) <br> epithelial cell proliferation <br> 9.651e-03","GO:0048729 (416) <br> tissue morphogenesis <br> 1.417e-02","GO:0030335 (424) <br> positive regulation of cell migration <br> 1.428e-02","GO:0030855 (412) <br> epithelial cell differentiation <br> 2.051e-02","GO:0072001 (202) <br> renal system development <br> 2.269e-02","GO:0051716 (4337) <br> cellular response to stimulus <br> 2.346e-02","GO:0010720 (408) <br> positive regulation of cell development <br> 2.963e-02","GO:0001501 (331) <br> skeletal system development <br> 3.360e-02","GO:0035272 (34) <br> exocrine system development <br> 3.869e-02","GO:0001649 (144) <br> osteoblast differentiation <br> 3.945e-02","GO:0045229 (192) <br> external encapsulating structure organization <br> 4.022e-02","GO:0030198 (192) <br> extracellular matrix organization <br> 4.022e-02","GO:0051241 (842) <br> negative regulation of multicellular organismal process <br> 4.309e-02","GO:0006935 (288) <br> chemotaxis <br> 4.537e-02","GO:0042330 (288) <br> taxis <br> 4.537e-02","GO:0043062 (193) <br> extracellular structure organization <br> 4.628e-02"],"key":["GO:0032501","GO:0048731","GO:0048513","GO:0007275","GO:0048856","GO:0032502","GO:0009653","GO:0009888","GO:0007154","GO:0051239","GO:0023052","GO:0048869","GO:0030154","GO:0051240","GO:0001503","GO:0048870","GO:0009887","GO:0003008","GO:0007399","GO:0007155","GO:0007166","GO:0030030","GO:0007423","GO:0120036","GO:0048468","GO:0050793","GO:0007267","GO:0016477","GO:0040011","GO:0007165","GO:0051094","GO:0040012","GO:2000026","GO:0099536","GO:0045595","GO:0001654","GO:0022008","GO:0048699","GO:0150063","GO:0040017","GO:0060429","GO:0099537","GO:0043010","GO:0030182","GO:0048880","GO:0042127","GO:0007268","GO:0098916","GO:0048646","GO:0098609","GO:0008285","GO:0045597","GO:0048666","GO:0034330","GO:0035239","GO:0035295","GO:0000902","GO:2000145","GO:0030334","GO:0031175","GO:0061564","GO:0050877","GO:0007409","GO:0072359","GO:0008283","GO:2000147","GO:0048667","GO:0051093","GO:0007411","GO:0097485","GO:0042221","GO:0048858","GO:0048812","GO:0120039","GO:0065008","GO:0050673","GO:0048729","GO:0030335","GO:0030855","GO:0072001","GO:0051716","GO:0010720","GO:0001501","GO:0035272","GO:0001649","GO:0045229","GO:0030198","GO:0051241","GO:0006935","GO:0042330","GO:0043062"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.80000000000000004,"size":[22.242643247049507,21.483223721485867,21.023602379512834,21.736886376670306,22.04139247052456,22.137227654716124,20.938855216613359,20.440874838753594,21.997367787072275,21.216785622972814,21.995089316068526,21.590522455953501,21.590522455953501,20.418116747068208,17.833997818106482,20.312887467074798,19.35721517792096,20.156402611917862,20.795137286011101,19.799578598519425,20.943840662476873,20.303756859699785,18.376327031990002,20.278675131535419,21.175372122803719,20.935521567907546,19.634837063586978,20.160747292972655,19.805133382726073,21.869149810978495,20.037002831411268,19.597002532832047,20.134480008492027,18.909031525020065,20.269322285267041,17.716825423275118,20.40355272013446,20.159300525236688,17.73812457464911,18.684975199949601,19.613937636500776,18.804674731019031,17.487078621002347,20.075867674360953,17.752171147795465,20.289289629490739,18.786572377354378,18.786572377354378,19.626512762939509,19.173862084678287,18.909031525020065,19.411258854120515,19.818013086625335,19.016167605077886,19.182315738216474,19.536154907814417,19.507032116360538,19.536154907814417,19.468066110795345,19.634837063586978,18.293655046624814,19.288106555815069,18.068793633600283,19.620238585741966,20.592815733346068,18.64159433804085,18.673263789889926,19.382022958502542,16.512933881310833,16.512933881310833,21.283129153583115,18.861212263443722,18.804674731019031,18.843766481910556,21.045348189153856,18.141842008583492,18.567888512558845,18.601078033995055,18.551025120052781,17.248918381679793,22.208874957271156,18.533977647304045,18.16366274984625,13.239545032802289,16.584326392579889,17.151327431175957,17.151327431175957,19.750533412464982,17.91189747818218,17.91189747818218,17.161345222930262],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"GO:BP","legendgroup":"GO:BP","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[30.692034030996147,30.778065386693033,37.121612702342979,31.238586173070495,31.807911321064605,31.825623659002204,38.118564294830449,32.207704091656026,31.972383030485126,34.654537061035164,34.887327788214975,37.06594535453911,37.053293684583686,39.277457262747347,37.045702682610433,37.048233016601515,37.060884686556939,37.055824018574768,38.191943980571914,37.073536356512363,37.068475688530192],"y":[17,17,17,17,17,17,17,8.9521446465123091,6.1071660524798714,4.1466497196230714,4.1023518178155154,3.7332469376768085,2.6123521453403176,2.3178993282682638,1.9854941868483298,1.9854941868483298,1.9854941868483298,1.9854941868483298,1.9591784051610854,1.4986558715297882,1.4986558715297882],"text":["GO:0005576 (917) <br> extracellular region <br> 1.234e-36","GO:0005615 (806) <br> extracellular space <br> 3.851e-34","GO:0071944 (3030) <br> cell periphery <br> 3.979e-29","GO:0005886 (2763) <br> plasma membrane <br> 1.779e-27","GO:0009897 (324) <br> external side of plasma membrane <br> 2.915e-23","GO:0009986 (630) <br> cell surface <br> 1.051e-22","GO:0098552 (499) <br> side of membrane <br> 6.311e-22","GO:0019814 (17) <br> immunoglobulin complex <br> 1.116e-09","GO:0016020 (4488) <br> membrane <br> 7.813e-07","GO:0042571 (11) <br> immunoglobulin complex, circulating <br> 7.134e-05","GO:0043235 (234) <br> receptor complex <br> 7.900e-05","GO:0071753 (9) <br> IgM immunoglobulin complex <br> 1.848e-04","GO:0071748 (8) <br> monomeric IgA immunoglobulin complex <br> 2.441e-03","GO:0170014 (6) <br> ankyrin-1 complex <br> 4.810e-03","GO:0071745 (9) <br> IgA immunoglobulin complex <br> 1.034e-02","GO:0071746 (9) <br> IgA immunoglobulin complex, circulating <br> 1.034e-02","GO:0071751 (9) <br> secretory IgA immunoglobulin complex <br> 1.034e-02","GO:0071749 (9) <br> polymeric IgA immunoglobulin complex <br> 1.034e-02","GO:0098590 (773) <br> plasma membrane region <br> 1.099e-02","GO:0071756 (7) <br> pentameric IgM immunoglobulin complex <br> 3.172e-02","GO:0071754 (7) <br> IgM immunoglobulin complex, circulating <br> 3.172e-02"],"key":["GO:0005576","GO:0005615","GO:0071944","GO:0005886","GO:0009897","GO:0009986","GO:0098552","GO:0019814","GO:0016020","GO:0042571","GO:0043235","GO:0071753","GO:0071748","GO:0170014","GO:0071745","GO:0071746","GO:0071751","GO:0071749","GO:0098590","GO:0071756","GO:0071754"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.80000000000000004,"size":[19.887763387769468,19.679799255456206,21.69987433543346,21.566592355033549,18.125278448744737,19.274951938102795,18.88188898526295,11.109665636853626,22.256717646743645,9.371655570012086,17.527665701548681,8.3532402425319212,7.6320797849196316,3.7795275590551185,8.3532402425319212,8.3532402425319212,8.3532402425319212,8.3532402425319212,19.611831332679444,6.5996298257682193,6.5996298257682193],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[37.121612702342979,31.238586173070495,34.472353013677044,30.692034030996147,30.778065386693033,34.78611442857158,34.78358409458049,38.76379946255711,32.501222834621878,38.191943980571914,35.302302562752907,34.791175096553751,32.741604563774949,32.488571164666453,32.503753168612967,37.655513174461902,37.389828105397982,36.033569086176456,32.341811793183531,34.021953563263928,35.530032621950546,38.298218008197473,35.532562955941636],"y":[17,17,8.2260632847445994,7.9119604094445615,7.8279055842298035,7.6052840478214048,7.0653924929033893,6.1327947770761702,5.4079412591504035,5.2587762127029452,4.6139191908419948,4.4395607985362293,4.3305697254076563,4.2610502226644318,4.2565365619213749,4.1661795324505926,3.715612311093202,2.7640036490245441,2.5791944116820136,1.9201502931132151,1.850649100572195,1.4355238999297104,1.3069244560821554],"text":["GO:0071944 (3030) <br> cell periphery <br> 1.453e-25","GO:0005886 (2763) <br> plasma membrane <br> 9.715e-18","GO:0036477 (718) <br> somatodendritic compartment <br> 5.942e-09","GO:0005576 (917) <br> extracellular region <br> 1.225e-08","GO:0005615 (806) <br> extracellular space <br> 1.486e-08","GO:0043005 (933) <br> neuron projection <br> 2.482e-08","GO:0042995 (1514) <br> cell projection <br> 8.602e-08","GO:0120025 (1493) <br> plasma membrane bounded cell projection <br> 7.366e-07","GO:0030424 (485) <br> axon <br> 3.909e-06","GO:0098590 (773) <br> plasma membrane region <br> 5.511e-06","GO:0044297 (564) <br> cell body <br> 2.433e-05","GO:0043025 (490) <br> neuronal cell body <br> 3.634e-05","GO:0031012 (237) <br> extracellular matrix <br> 4.671e-05","GO:0030312 (238) <br> external encapsulating structure <br> 5.482e-05","GO:0030425 (494) <br> dendrite <br> 5.539e-05","GO:0097447 (496) <br> dendritic tree <br> 6.821e-05","GO:0097060 (260) <br> synaptic membrane <br> 1.925e-04","GO:0062023 (192) <br> collagen-containing extracellular matrix <br> 1.722e-03","GO:0030054 (1442) <br> cell junction <br> 2.635e-03","GO:0034703 (94) <br> cation channel complex <br> 1.202e-02","GO:0045202 (1073) <br> synapse <br> 1.410e-02","GO:0098797 (340) <br> plasma membrane protein complex <br> 3.668e-02","GO:0045211 (159) <br> postsynaptic membrane <br> 4.933e-02"],"key":["GO:0071944","GO:0005886","GO:0036477","GO:0005576","GO:0005615","GO:0043005","GO:0042995","GO:0120025","GO:0030424","GO:0098590","GO:0044297","GO:0043025","GO:0031012","GO:0030312","GO:0030425","GO:0097447","GO:0097060","GO:0062023","GO:0030054","GO:0034703","GO:0045202","GO:0098797","GO:0045211"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.80000000000000004,"size":[21.69987433543346,21.566592355033549,19.491110782627455,19.887763387769468,19.679799255456206,19.915440605596359,20.67164870184585,20.65030511192321,18.833203192945678,19.611831332679444,19.089612262880703,18.850768575325436,17.551547639234933,17.55943205209639,18.864677776311257,18.871585430708162,17.723956051364301,17.151327431175957,20.5970770554677,15.694133435847569,20.137422405435927,18.211693797899965,16.782419293748923],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"GO:CC","legendgroup":"GO:CC","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2.9183733560476282,14.735789185520002,21.720992590609537,4.84619841750298,25.723380054155946,11.082535394520621,4.9170371336719434,5.1801523651566628,12.595447975557761,4.8537882799496543],"y":[17,17,17,17,9.4608160819868043,5.2709748662451306,4.5956463344282925,2.5826349887598443,2.3288372408459854,1.6353408837341661],"text":["GO:0003823 (118) <br> antigen binding <br> 9.804e-42","GO:0038023 (497) <br> signaling receptor activity <br> 1.618e-17","GO:0060089 (497) <br> molecular transducer activity <br> 1.618e-17","GO:0004888 (371) <br> transmembrane signaling receptor activity <br> 3.811e-17","GO:0140375 (101) <br> immune receptor activity <br> 3.461e-10","GO:0023023 (18) <br> MHC protein complex binding <br> 5.358e-06","GO:0004930 (146) <br> G protein-coupled receptor activity <br> 2.537e-05","GO:0005102 (935) <br> signaling receptor binding <br> 2.614e-03","GO:0032394 (6) <br> MHC class Ib receptor activity <br> 4.690e-03","GO:0004896 (76) <br> cytokine receptor activity <br> 2.316e-02"],"key":["GO:0003823","GO:0038023","GO:0060089","GO:0004888","GO:0140375","GO:0023023","GO:0004930","GO:0005102","GO:0032394","GO:0004896"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.80000000000000004,"size":[16.176679709868722,18.875027639290309,18.875027639290309,18.36682260038814,15.848640317190892,11.308129957575494,16.612083865855723,19.918863535097596,3.7795275590551185,15.22467129476737],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[21.720992590609537,14.735789185520002,4.84619841750298,20.701421068606244,5.1801523651566628,19.019001559593374,11.388659846536497,11.391189800685387,5.220631631538927],"y":[6.3679154715236717,6.3679154715236717,4.0962907706029634,3.208738536497548,2.5892368078313988,2.0525641527918861,1.7462196439669426,1.7104717571550645,1.5577642575048973],"text":["GO:0060089 (497) <br> molecular transducer activity <br> 4.286e-07","GO:0038023 (497) <br> signaling receptor activity <br> 4.286e-07","GO:0004888 (371) <br> transmembrane signaling receptor activity <br> 8.011e-05","GO:0050839 (190) <br> cell adhesion molecule binding <br> 6.184e-04","GO:0005102 (935) <br> signaling receptor binding <br> 2.575e-03","GO:0048018 (173) <br> receptor ligand activity <br> 8.860e-03","GO:0030545 (198) <br> signaling receptor regulator activity <br> 1.794e-02","GO:0030546 (178) <br> signaling receptor activator activity <br> 1.948e-02","GO:0005125 (79) <br> cytokine activity <br> 2.768e-02"],"key":["GO:0060089","GO:0038023","GO:0004888","GO:0050839","GO:0005102","GO:0048018","GO:0030545","GO:0030546","GO:0005125"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.80000000000000004,"size":[18.875027639290309,18.875027639290309,18.36682260038814,17.131111515468088,19.918863535097596,16.948769257936256,17.210559172633918,17.004461977093495,15.311599489713084],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"GO:MF","legendgroup":"GO:MF","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[151.26283534053553,173.14846196909176,153.60040544859078,168.59475396638675,170.90196602109063,179.82976354417175,152.35825509896401,151.39691674283739,150.74168875800373,156.50718905698417,174.83080409231334,171.28397152576198,156.66656883707884,153.95205289991077],"y":[6.1593833935446201,4.5399602816382938,4.0822735272308339,3.8003952908174141,3.3020273936617639,3.1568835749507262,2.9297785564520997,2.4737413856031742,2.4637013179756804,2.1191652066738942,2.007147198552754,1.9265265161066754,1.7220116710775515,1.6295857405621581],"text":["HP:0001287 (71) <br> Meningitis <br> 6.928e-07","HP:0025142 (981) <br> Constitutional symptom <br> 2.884e-05","HP:0002633 (67) <br> Vasculitis <br> 8.274e-05","HP:0011450 (98) <br> Unusual CNS infection <br> 1.583e-04","HP:0012378 (385) <br> Fatigue <br> 4.989e-04","HP:0032158 (105) <br> Unusual infection by anatomical site <br> 6.968e-04","HP:0001945 (372) <br> Fever <br> 1.175e-03","HP:0001369 (204) <br> Arthritis <br> 3.359e-03","HP:0000988 (148) <br> Skin rash <br> 3.438e-03","HP:0004370 (463) <br> Abnormality of temperature regulation <br> 7.600e-03","HP:0030057 (137) <br> Autoimmune antibody positivity <br> 9.837e-03","HP:0012531 (667) <br> Pain <br> 1.184e-02","HP:0004444 (7) <br> Spherocytosis <br> 1.897e-02","HP:0002829 (171) <br> Arthralgia <br> 2.346e-02"],"key":["HP:0001287","HP:0025142","HP:0002633","HP:0011450","HP:0012378","HP:0032158","HP:0001945","HP:0001369","HP:0000988","HP:0004370","HP:0030057","HP:0012531","HP:0004444","HP:0002829"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(153,0,153,1)","opacity":0.80000000000000004,"size":[15.070244340493012,19.995443519066075,14.936973221284303,15.784019234776659,18.432179820768962,15.931371709132424,18.371581977412763,17.267773962595452,16.639404978777065,18.753443017852959,16.483539132242655,19.369670494337328,6.5996298257682193,16.92597234664338],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(153,0,153,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"HP","legendgroup":"HP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[113.29837139387047,113.68609645804942,113.68863060879569,113.45295458939282,113.09563933416906,113.67595985506436,113.67342570431809,113.56699137297485,113.55938892073605],"y":[9.8827772537367853,4.10378679046093,4.10378679046093,3.5718511166841762,3.2503999938725205,2.5937545282510244,2.4260075542205377,2.1880841329126692,1.558293534936223],"text":["KEGG:04640 (79) <br> Hematopoietic cell lineage <br> 1.310e-10","KEGG:05330 (43) <br> Allograft rejection <br> 7.874e-05","KEGG:05332 (43) <br> Graft-versus-host disease <br> 7.874e-05","KEGG:04940 (46) <br> Type I diabetes mellitus <br> 2.680e-04","KEGG:04060 (183) <br> Cytokine-cytokine receptor interaction <br> 5.618e-04","KEGG:05320 (41) <br> Autoimmune thyroid disease <br> 2.548e-03","KEGG:05310 (18) <br> Asthma <br> 3.750e-03","KEGG:05150 (38) <br> Staphylococcus aureus infection <br> 6.485e-03","KEGG:05144 (48) <br> Malaria <br> 2.765e-02"],"key":["KEGG:04640","KEGG:05330","KEGG:05332","KEGG:04940","KEGG:04060","KEGG:05320","KEGG:05310","KEGG:05150","KEGG:05144"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.80000000000000004,"size":[15.311599489713084,13.859598391479308,13.859598391479308,14.030735967020609,17.058387800350264,13.736966531684869,11.308129957575494,13.538129342461552,14.137279201763615],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[113.09563933416906,113.12098084163175],"y":[8.278493169632279,2.2542968929461344],"text":["KEGG:04060 (183) <br> Cytokine-cytokine receptor interaction <br> 5.266e-09","KEGG:04080 (123) <br> Neuroactive ligand-receptor interaction <br> 5.568e-03"],"key":["KEGG:04060","KEGG:04080"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.80000000000000004,"size":[17.058387800350264,16.262734076959585],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"KEGG","legendgroup":"KEGG","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[116.95008211942445,117.63601749554176,117.0817007155798,119.97984095784665,116.71468732245431,117.18041466269632,119.85581574223873,117.25887959501969,116.8614926797045,118.18527202180543,118.15742962646488,119.85834686908785,117.62842411499433,117.22597494598087,119.63054545266515,117.63095524184348,117.71701355471428,119.72419714608337,118.06884018674494,120.16967547153224,118.19033427550373,117.63348636869262,117.71448242786514,118.15489849961574,116.63622239013092,117.16522790160147,117.87141229251191,117.58033270486065,120.08361715866145,118.86108289052618,117.12219874516606,117.30443987830424,119.10407106804375],"y":[17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,15.952903201231955,13.582455822504263,11.381195510902984,10.628791935559628,5.7155297336165916,4.5336787892591577,4.2492285304998685,3.250381153512282,2.6366980613911615,2.5644767747167063,2.5272590127291945,2.3942419858085007,2.0443961764934628],"text":["REAC:R-MMU-5690714 (47) <br> CD22 mediated BCR regulation <br> 2.481e-38","REAC:R-MMU-2029481 (55) <br> FCGR activation <br> 1.033e-37","REAC:R-MMU-202733 (116) <br> Cell surface interactions at the vascular wall <br> 7.767e-36","REAC:R-MMU-2168880 (58) <br> Scavenging of heme from plasma <br> 9.085e-36","REAC:R-MMU-983695 (64) <br> Antigen activates B Cell Receptor (BCR) leading to generation of second messengers <br> 1.365e-35","REAC:R-MMU-173623 (48) <br> Classical antibody-mediated complement activation <br> 1.533e-35","REAC:R-MMU-2730905 (54) <br> Role of LAT2/NTAL/LAB on calcium mobilization <br> 6.948e-35","REAC:R-MMU-166786 (50) <br> Creation of C4 and C2 activators <br> 4.550e-34","REAC:R-MMU-2173782 (65) <br> Binding and Uptake of Ligands by Scavenger Receptors <br> 1.586e-33","REAC:R-MMU-166663 (55) <br> Initial triggering of complement <br> 1.359e-32","REAC:R-MMU-198933 (120) <br> Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell <br> 2.527e-32","REAC:R-MMU-2029485 (62) <br> Role of phospholipids in phagocytosis <br> 5.622e-32","REAC:R-MMU-2871809 (66) <br> FCERI mediated Ca+2 mobilization <br> 4.217e-30","REAC:R-MMU-166658 (66) <br> Complement cascade <br> 4.217e-30","REAC:R-MMU-977606 (63) <br> Regulation of Complement cascade <br> 5.297e-30","REAC:R-MMU-2871796 (68) <br> FCERI mediated MAPK activation <br> 7.386e-28","REAC:R-MMU-2029480 (113) <br> Fcgamma receptor (FCGR) dependent phagocytosis <br> 5.551e-21","REAC:R-MMU-2029482 (94) <br> Regulation of actin dynamics for phagocytic cup formation <br> 5.465e-20","REAC:R-MMU-109582 (427) <br> Hemostasis <br> 3.487e-19","REAC:R-MMU-983705 (137) <br> Signaling by the B Cell Receptor (BCR) <br> 1.589e-17","REAC:R-MMU-168249 (769) <br> Innate Immune System <br> 1.115e-16","REAC:R-MMU-2871837 (113) <br> FCERI mediated NF-kB activation <br> 2.615e-14","REAC:R-MMU-2454202 (152) <br> Fc epsilon receptor (FCERI) signaling <br> 4.157e-12","REAC:R-MMU-168256 (1330) <br> Immune System <br> 2.351e-11","REAC:R-MMU-1280218 (625) <br> Adaptive Immune System <br> 1.925e-06","REAC:R-MMU-373076 (123) <br> Class A/1 (Rhodopsin-like receptors) <br> 2.926e-05","REAC:R-MMU-500792 (155) <br> GPCR ligand binding <br> 5.633e-05","REAC:R-MMU-1247673 (7) <br> Erythrocytes take up oxygen and release carbon dioxide <br> 5.618e-04","REAC:R-MMU-372790 (325) <br> Signaling by GPCR <br> 2.308e-03","REAC:R-MMU-6798695 (402) <br> Neutrophil degranulation <br> 2.726e-03","REAC:R-MMU-380108 (30) <br> Chemokine receptors bind chemokines <br> 2.970e-03","REAC:R-MMU-2172127 (52) <br> DAP12 interactions <br> 4.034e-03","REAC:R-MMU-375276 (67) <br> Peptide ligand-binding receptors <br> 9.028e-03"],"key":["REAC:R-MMU-5690714","REAC:R-MMU-2029481","REAC:R-MMU-202733","REAC:R-MMU-2168880","REAC:R-MMU-983695","REAC:R-MMU-173623","REAC:R-MMU-2730905","REAC:R-MMU-166786","REAC:R-MMU-2173782","REAC:R-MMU-166663","REAC:R-MMU-198933","REAC:R-MMU-2029485","REAC:R-MMU-2871809","REAC:R-MMU-166658","REAC:R-MMU-977606","REAC:R-MMU-2871796","REAC:R-MMU-2029480","REAC:R-MMU-2029482","REAC:R-MMU-109582","REAC:R-MMU-983705","REAC:R-MMU-168249","REAC:R-MMU-2871837","REAC:R-MMU-2454202","REAC:R-MMU-168256","REAC:R-MMU-1280218","REAC:R-MMU-373076","REAC:R-MMU-500792","REAC:R-MMU-1247673","REAC:R-MMU-372790","REAC:R-MMU-6798695","REAC:R-MMU-380108","REAC:R-MMU-2172127","REAC:R-MMU-375276"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(51,102,204,1)","opacity":0.80000000000000004,"size":[14.084712022968755,14.470943792135303,16.141058272199061,14.598327640764314,14.830553653723618,14.137279201763615,14.426579176757247,14.238452775399061,14.866685592072582,14.470943792135303,16.211602933105151,14.756193341946302,14.902151176257419,14.902151176257419,14.793731340751455,14.971173415121687,16.086257540638403,15.694133435847569,18.61334407627615,16.483539132242655,19.603375960409068,16.086257540638403,16.692789017537386,20.472598917522426,19.261681190677145,16.262734076959585,16.731773338198007,6.5996298257682193,18.130818742330046,18.508052744714877,12.891853572650168,14.334744307141492,14.936973221284303],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(51,102,204,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"REAC","legendgroup":"REAC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[120.08361715866145,117.86888116566276,117.62083073444691],"y":[2.6185501024137525,2.6012785090433437,1.9246802565760548],"text":["REAC:R-MMU-372790 (325) <br> Signaling by GPCR <br> 2.407e-03","REAC:R-MMU-388396 (309) <br> GPCR downstream signalling <br> 2.505e-03","REAC:R-MMU-1474244 (144) <br> Extracellular matrix organization <br> 1.189e-02"],"key":["REAC:R-MMU-372790","REAC:R-MMU-388396","REAC:R-MMU-1474244"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(51,102,204,1)","opacity":0.80000000000000004,"size":[18.130818742330046,18.039786376536178,16.584326392579889],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(51,102,204,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"REAC","legendgroup":"REAC","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[130.77632866249402,129.70601441936057,132.02629375967589,129.74649911886445,130.78138924993203,131.391190036209],"y":[4.2831672363890441,3.6183161416480933,2.933441903609205,2.2507425761066968,1.7593054879735306,1.67596210506944],"text":["TF:M01172 (2832) <br> Factor: PU.1; motif: NNNNYYYACTTCCTCTTTY <br> 5.210e-05","TF:M01808_1 (4375) <br> Factor: NMYC; motif: CAYCTG; match class: 1 <br> 2.408e-04","TF:M00749 (5845) <br> Factor: SREBP-1; motif: CACSCCA <br> 1.166e-03","TF:M07107_1 (4242) <br> Factor: LXR-alpha:RXR-alpha; motif: TGACCTNNAGTRACCYNNN; match class: 1 <br> 5.614e-03","TF:M00658 (3135) <br> Factor: PU.1; motif: WGAGGAAG <br> 1.741e-02","TF:M00701_1 (740) <br> Factor: SMAD3; motif: TGTCTGTCT; match class: 1 <br> 2.109e-02"],"key":["TF:M01172","TF:M01808_1","TF:M00749","TF:M07107_1","TF:M00658","TF:M00701_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.80000000000000004,"size":[21.60232909577935,22.221081677590909,22.621930759440549,22.177847603052335,21.748845798378699,19.540585030821266],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"TF","legendgroup":"TF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[131.91496083604028,126.53555638946405,126.92775191590776,133.15480475834616,132.89924509272802],"y":[5.6443571256505711,2.5039362007303048,1.6902401559816187,1.537881114177877,1.3474809763331441],"text":["TF:M10376 (3559) <br> Factor: Sp5; motif: RNGGRGGNGGRGNNGGGGGAGGRG <br> 2.268e-06","TF:M01587 (1740) <br> Factor: FPM315; motif: SRGGGAGGAGGN <br> 3.134e-03","TF:M07040_1 (6086) <br> Factor: GKLF; motif: NNRRGRRNGNSNNN; match class: 1 <br> 2.041e-02","TF:M07436 (5624) <br> Factor: WT1; motif: NNGGGNGGGSGN <br> 2.898e-02","TF:M00706_1 (957) <br> Factor: TFII-I; motif: RGAGGKAGG; match class: 1 <br> 4.493e-02"],"key":["TF:M10376","TF:M01587","TF:M07040_1","TF:M07436","TF:M00706_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.80000000000000004,"size":[21.930034280169245,20.882795697777869,22.677165354330711,22.569088928506897,19.95599315104813],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"TF","legendgroup":"TF","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[199.57038207331098,199.61613888135815],"y":[4.5897050401958976,3.4506602560976933],"text":["WP:WP1396 (118) <br> GPCRs non odorant  <br> 2.572e-05","WP:WP189 (73) <br> GPCRs class A rhodopsin like <br> 3.543e-04"],"key":["WP:WP1396","WP:WP189"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,153,198,1)","opacity":0.80000000000000004,"size":[16.176679709868722,15.133535441877722],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,153,198,1)"}},"hoveron":"points","set":"SharedDataba555065","name":"WP","legendgroup":"WP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,27.673974702951323],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2,27.673974702951323],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[30.203679570716751,40.378152548869295],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[30.203679570716751,40.378152548869295],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[42.907857416634727,109.92732847834419],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[42.907857416634727,109.92732847834419],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[112.45703334610963,113.89896512073591],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[112.45703334610963,113.89896512073591],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[116.42866998850134,120.93154465312381],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(51,102,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[116.42866998850134,120.93154465312381],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(51,102,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[123.46124952088923,134.33139133767727],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[123.46124952088923,134.33139133767727],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[136.86109620544272,142.02422384055194],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[136.86109620544272,142.02422384055194],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[144.55392870831739,146.27159831353009],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[144.55392870831739,146.27159831353009],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[148.80130318129554,196.95170563434269],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[148.80130318129554,196.95170563434269],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[199.4814105021081,200.00252970486778],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(0,153,198,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[199.4814105021081,200.00252970486778],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(0,153,198,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[180],"y":[16.199999999999999],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.559055118110237,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.199999999999999],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.559055118110237,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.75590551181102372,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.75590551181102372,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.283520132835207,"r":6.6417600664176026,"b":55.878166138820198,"l":61.037775010377771},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"tickvals":[14.836987351475663,35.29091605979302,76.417592947489453,113.17799923342277,118.68010732081255,128.89632042928326,139.44266002299733,145.41276351092375,172.87650440781911,199.74197010348794],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.62681610626816},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.13283520132835203,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0.57420999570051356,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.393109173931101},{"text":"downreg_genes","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"upreg_genes","x":0.5,"y":0.42579000429948644,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0.57420999570051356,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.973017849730184,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":0.42579000429948644},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.973017849730184,"yanchor":0.42579000429948644,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"tickvals":[14.836987351475663,35.29091605979302,76.417592947489453,113.17799923342277,118.68010732081255,128.89632042928326,139.44266002299733,145.41276351092375,172.87650440781911,199.74197010348794],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.62681610626816},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.13283520132835203,"showgrid":false,"domain":[0,1],"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"yaxis2":{"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176002,"showgrid":false,"domain":[0,0.42579000429948644],"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"3743fa5de9a597":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"3743fa4f776329":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa3b4155a0":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa72258b7f":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa273db0f8":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa5d2b5cc4":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa3aef5ae2":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa7b3bb825":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa6e84cb78":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa63457cdb":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa543a8b31":{"x":{},"xend":{},"y":{},"yend":{}},"3743fa3697aea6":{"x":{},"y":{}},"3743fa45f37745":{"yintercept":{}}},"cur_data":"3743fa5de9a597","visdat":{"3743fa5de9a597":["function (y) ","x"],"3743fa4f776329":["function (y) ","x"],"3743fa3b4155a0":["function (y) ","x"],"3743fa72258b7f":["function (y) ","x"],"3743fa273db0f8":["function (y) ","x"],"3743fa5d2b5cc4":["function (y) ","x"],"3743fa3aef5ae2":["function (y) ","x"],"3743fa7b3bb825":["function (y) ","x"],"3743fa6e84cb78":["function (y) ","x"],"3743fa63457cdb":["function (y) ","x"],"3743fa543a8b31":["function (y) ","x"],"3743fa3697aea6":["function (y) ","x"],"3743fa45f37745":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDataba555065"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<ol start="7" type="a">
<li>Output the <code>get_version_info()</code> to save the versions of the databases that were used for the analysis!</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_version_info</span>(<span class="st">'mmusculus'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$biomart
[1] "Ensembl"

$biomart_version
[1] "112"

$display_name
[1] "Mouse"

$genebuild
[1] "GRCm39"

$gprofiler_version
[1] "e112_eg59_p19_25aa4782"

$organism
[1] "mmusculus"

$sources
$sources$CORUM
$sources$CORUM$name
[1] "CORUM protein complexes"

$sources$CORUM$version
[1] "28.11.2022 Corum 4.1"


$sources$`GO:BP`
$sources$`GO:BP`$name
[1] "biological process"

$sources$`GO:BP`$version
[1] "annotations: BioMart\nclasses: releases/2024-10-27"


$sources$`GO:CC`
$sources$`GO:CC`$name
[1] "cellular component"

$sources$`GO:CC`$version
[1] "annotations: BioMart\nclasses: releases/2024-10-27"


$sources$`GO:MF`
$sources$`GO:MF`$name
[1] "molecular function"

$sources$`GO:MF`$version
[1] "annotations: BioMart\nclasses: releases/2024-10-27"


$sources$HP
$sources$HP$name
[1] "Human Phenotype Ontology"

$sources$HP$version
[1] "annotations: 02.2025\nclasses: None"


$sources$KEGG
$sources$KEGG$name
[1] "Kyoto Encyclopedia of Genes and Genomes"

$sources$KEGG$version
[1] "KEGG FTP Release 2024-01-22"


$sources$MIRNA
$sources$MIRNA$name
[1] "miRTarBase"

$sources$MIRNA$version
[1] "Release 9.0"


$sources$REAC
$sources$REAC$name
[1] "Reactome"

$sources$REAC$version
[1] "annotations: BioMart\nclasses: 2025-2-3"


$sources$TF
$sources$TF$name
[1] "Transfac"

$sources$TF$version
[1] "annotations: TRANSFAC Release 2023.2\nclasses: v2"


$sources$WP
$sources$WP$name
[1] "WikiPathways"

$sources$WP$version
[1] "20250110"



$taxonomy_id
[1] "10090"</code></pre>
</div>
</div>
</section>
</section>
<section id="gsea" class="level1">
<h1>GSEA</h1>
<ul>
<li><p>We will use the <a href="https://bioconductor.org/packages/release/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html"><strong>fgsea</strong></a> R package to perform GSEA on the output table of differentially expressed genes from DESeq2.</p></li>
<li><p>The major aims of this exercise is to get you used to reading online tool documentation, playing with different parameters of the relevant functions and interpreting outputs.</p></li>
</ul>
<section id="load-libraries.-1" class="level3">
<h3 class="anchored" data-anchor-id="load-libraries.-1">1. Load libraries.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-output-table-of-differentially-expressed-genes-from-deseq2.-1" class="level3">
<h3 class="anchored" data-anchor-id="read-in-output-table-of-differentially-expressed-genes-from-deseq2.-1">2. Read in output table of differentially expressed genes from DESeq2.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>all_genes_df <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"dds_results.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 19053 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (1): gene_id
dbl (6): baseMean, log2FoldChange, lfcSE, stat, pvalue, padj

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
</section>
<section id="create-input-gene-list." class="level3">
<h3 class="anchored" data-anchor-id="create-input-gene-list.">3. Create input gene list.</h3>
<ul>
<li>Create <strong>all_genes_tested_df</strong> object containing table of all genes tested in the differential expression analysis.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>all_genes_tested_df <span class="ot">&lt;-</span> all_genes_df <span class="sc">%&gt;%</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(padj))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>All we’ll care about later on is the gene identifier and a gene-level statistic to rank the genes i.e.&nbsp;log2FoldChange. Simplify table by getting just those columns.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>all_genes_tested_df <span class="ot">&lt;-</span> all_genes_tested_df <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id,log2FoldChange)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Check whether you have genes with multiple entries.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">any</span>(<span class="fu">duplicated</span>(all_genes_tested_df<span class="sc">$</span>gene_id)) <span class="co"># Gene ids are unique so we'll just proceed</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] FALSE</code></pre>
</div>
</div>
</section>
<section id="obtain-list-of-gene-sets-pathways" class="level3">
<h3 class="anchored" data-anchor-id="obtain-list-of-gene-sets-pathways">4. Obtain list of gene sets / pathways</h3>
<ol type="a">
<li>The <code>fgsea()</code> function requires (1) a list of gene sets / pathways to test enrichment of, and (2) a named vector of gene-level statistic (log2FoldChange), where the names should be the same as the gene names in the gene sets / pathways. First, let’s create our named vector of log2FoldChange values.</li>
</ol>
<ul>
<li>See <code>?tibble::deframe</code> for help here. The <code>deframe()</code> function converts 2-column data frames to a named vector or list, using the first column as name and the second column as value.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>ranks <span class="ot">&lt;-</span> <span class="fu">deframe</span>(all_genes_tested_df)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ranks)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10-pathways_files/figure-html/unnamed-chunk-21-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ul>
<li>Sort the genes based on the ranking metric.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>ranks <span class="ot">&lt;-</span> <span class="fu">sort</span>(ranks, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ranks)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10-pathways_files/figure-html/unnamed-chunk-22-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ol start="2" type="a">
<li>Obtain gene sets / pathways. Let’s use the <a href="https://www.gsea-msigdb.org/gsea/msigdb/human/collections.jsp">Hallmark gene sets from MSigDB</a>. “Hallmark gene sets summarise and represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying overlaps between gene sets in other MSigDB collections and retaining genes that display coordinate expression.”</li>
</ol>
<p>We can either download GMT files directly from MSigDB or use the <a href="https://cran.r-project.org/web/packages/msigdbr/index.html">misgidbr</a> R package. We’ll do the latter. Check the <a href="https://cran.r-project.org/web/packages/msigdbr/vignettes/msigdbr-intro.html">documentation</a> for the functions we need.</p>
<ul>
<li>Use a helper function from misgidbr to show the available species.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">msigdbr_species</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 20 × 2
   species_name                    species_common_name                          
   &lt;chr&gt;                           &lt;chr&gt;                                        
 1 Anolis carolinensis             Carolina anole, green anole                  
 2 Bos taurus                      bovine, cattle, cow, dairy cow, domestic cat…
 3 Caenorhabditis elegans          &lt;NA&gt;                                         
 4 Canis lupus familiaris          dog, dogs                                    
 5 Danio rerio                     leopard danio, zebra danio, zebra fish, zebr…
 6 Drosophila melanogaster         fruit fly                                    
 7 Equus caballus                  domestic horse, equine, horse                
 8 Felis catus                     cat, cats, domestic cat                      
 9 Gallus gallus                   bantam, chicken, chickens, Gallus domesticus 
10 Homo sapiens                    human                                        
11 Macaca mulatta                  rhesus macaque, rhesus macaques, Rhesus monk…
12 Monodelphis domestica           gray short-tailed opossum                    
13 Mus musculus                    house mouse, mouse                           
14 Ornithorhynchus anatinus        duck-billed platypus, duckbill platypus, pla…
15 Pan troglodytes                 chimpanzee                                   
16 Rattus norvegicus               brown rat, Norway rat, rat, rats             
17 Saccharomyces cerevisiae        baker's yeast, brewer's yeast, S. cerevisiae 
18 Schizosaccharomyces pombe 972h- &lt;NA&gt;                                         
19 Sus scrofa                      pig, pigs, swine, wild boar                  
20 Xenopus tropicalis              tropical clawed frog, western clawed frog    </code></pre>
</div>
</div>
<ul>
<li>Retrieve the hallmark gene sets as <strong>h_gene_sets</strong> object.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>h_gene_sets <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"Mus musculus"</span>, <span class="co"># Either scientific or common name is acceptable</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">category =</span> <span class="st">"H"</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If you want all the gene sets:</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="co"># all_gene_sets = msigdbr(species = "mouse")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>What is the class of the output. Explore contents.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(h_gene_sets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "tbl_df"     "tbl"        "data.frame"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>h_gene_sets</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 7,384 × 18
   gs_cat gs_subcat gs_name               gene_symbol entrez_gene ensembl_gene  
   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;                 &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;         
 1 H      ""        HALLMARK_ADIPOGENESIS Abca1             11303 ENSMUSG000000…
 2 H      ""        HALLMARK_ADIPOGENESIS Abcb8             74610 ENSMUSG000000…
 3 H      ""        HALLMARK_ADIPOGENESIS Acaa2             52538 ENSMUSG000000…
 4 H      ""        HALLMARK_ADIPOGENESIS Acadl             11363 ENSMUSG000000…
 5 H      ""        HALLMARK_ADIPOGENESIS Acadm             11364 ENSMUSG000000…
 6 H      ""        HALLMARK_ADIPOGENESIS Acads             11409 ENSMUSG000000…
 7 H      ""        HALLMARK_ADIPOGENESIS Acly             104112 ENSMUSG000000…
 8 H      ""        HALLMARK_ADIPOGENESIS Aco2              11429 ENSMUSG000000…
 9 H      ""        HALLMARK_ADIPOGENESIS Acox1             11430 ENSMUSG000000…
10 H      ""        HALLMARK_ADIPOGENESIS Adcy6             11512 ENSMUSG000000…
# ℹ 7,374 more rows
# ℹ 12 more variables: human_gene_symbol &lt;chr&gt;, human_entrez_gene &lt;int&gt;,
#   human_ensembl_gene &lt;chr&gt;, gs_id &lt;chr&gt;, gs_pmid &lt;chr&gt;, gs_geoid &lt;chr&gt;,
#   gs_exact_source &lt;chr&gt;, gs_url &lt;chr&gt;, gs_description &lt;chr&gt;, taxon_id &lt;int&gt;,
#   ortholog_sources &lt;chr&gt;, num_ortholog_sources &lt;dbl&gt;</code></pre>
</div>
</div>
<ul>
<li>Remove genes in <code>h_gene_sets</code> that were not considered in the differential expression analysis.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>h_gene_sets_filtered <span class="ot">&lt;-</span> h_gene_sets <span class="sc">%&gt;%</span> </span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ensembl_gene <span class="sc">%in%</span> all_genes_tested_df<span class="sc">$</span>gene_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>The <code>fgsea()</code> function accepts a <strong>list</strong> of gene sets. Create this list from <strong>h_gene_sets</strong> using base function <code>split()</code>.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>hallmark_list <span class="ot">&lt;-</span> <span class="fu">split</span>(<span class="at">x =</span> h_gene_sets_filtered<span class="sc">$</span>ensembl_gene,</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">f =</span> h_gene_sets_filtered<span class="sc">$</span>gs_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="perform-gsea-using-fgsea." class="level3">
<h3 class="anchored" data-anchor-id="perform-gsea-using-fgsea.">5. Perform GSEA using <strong>fgsea</strong>.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">190</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span> <span class="fu">fgsea</span>(<span class="at">pathways=</span>hallmark_list, </span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">stats=</span>ranks, </span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">maxSize=</span><span class="dv">500</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.65% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fgseaRes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                        pathway        pval       padj    log2err         ES
1:        HALLMARK_ADIPOGENESIS 0.515320334 0.92021488 0.05960370  0.2933895
2: HALLMARK_ALLOGRAFT_REJECTION 0.002179026 0.01556447 0.43170770 -0.3987698
3:   HALLMARK_ANDROGEN_RESPONSE 0.048961424 0.20583832 0.24891111  0.4365222
4:        HALLMARK_ANGIOGENESIS 0.292808219 0.63653961 0.10063339  0.4599768
5:     HALLMARK_APICAL_JUNCTION 0.077474892 0.24714411 0.19189224  0.3842531
6:      HALLMARK_APICAL_SURFACE 0.694490818 0.96457058 0.05412006  0.3402518
          NES size
1:  0.9723040  182
2: -1.4330451  186
3:  1.3351808   93
4:  1.1243128   27
5:  1.2503215  157
6:  0.8696272   33
                                                                                                             leadingEdge
1: ENSMUSG00000000440,ENSMUSG00000025348,ENSMUSG00000019846,ENSMUSG00000048537,ENSMUSG00000005803,ENSMUSG00000015568,...
2: ENSMUSG00000031779,ENSMUSG00000023132,ENSMUSG00000029417,ENSMUSG00000035042,ENSMUSG00000053977,ENSMUSG00000004296,...
3: ENSMUSG00000044485,ENSMUSG00000059042,ENSMUSG00000063089,ENSMUSG00000055435,ENSMUSG00000033910,ENSMUSG00000032883,...
4: ENSMUSG00000026042,ENSMUSG00000029304,ENSMUSG00000015568,ENSMUSG00000048450,ENSMUSG00000025810,ENSMUSG00000032548,...
5: ENSMUSG00000061048,ENSMUSG00000050541,ENSMUSG00000040690,ENSMUSG00000062300,ENSMUSG00000026207,ENSMUSG00000029151,...
6: ENSMUSG00000059857,ENSMUSG00000078716,ENSMUSG00000044254,ENSMUSG00000039166,ENSMUSG00000031778,ENSMUSG00000064177,...</code></pre>
</div>
</div>
<ul>
<li>Convert result to tibble and arrange by normalised enrichment score</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>fgseaResTidy <span class="ot">&lt;-</span> fgseaRes <span class="sc">%&gt;%</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(NES))</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>fgseaResTidy</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 50 × 8
   pathway                    pval    padj log2err    ES   NES  size leadingEdge
   &lt;chr&gt;                     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt; &lt;list&gt;     
 1 HALLMARK_EPITHELIAL_ME… 2.51e-6 4.19e-5   0.627 0.588  1.89   137 &lt;chr [51]&gt; 
 2 HALLMARK_IL2_STAT5_SIG… 2.04e-8 1.02e-6   0.734 0.569  1.89   196 &lt;chr [76]&gt; 
 3 HALLMARK_UV_RESPONSE_DN 2.57e-3 1.61e-2   0.432 0.488  1.55   124 &lt;chr [36]&gt; 
 4 HALLMARK_TGF_BETA_SIGN… 3.33e-2 1.67e-1   0.322 0.525  1.45    50 &lt;chr [16]&gt; 
 5 HALLMARK_ESTROGEN_RESP… 1.15e-2 6.39e-2   0.381 0.431  1.40   160 &lt;chr [39]&gt; 
 6 HALLMARK_PANCREAS_BETA… 7.91e-2 2.47e-1   0.211 0.628  1.39    17 &lt;chr [5]&gt;  
 7 HALLMARK_KRAS_SIGNALIN… 4.94e-2 2.06e-1   0.249 0.431  1.34   107 &lt;chr [35]&gt; 
 8 HALLMARK_ANDROGEN_RESP… 4.90e-2 2.06e-1   0.249 0.437  1.34    93 &lt;chr [16]&gt; 
 9 HALLMARK_INFLAMMATORY_… 5.40e-2 2.08e-1   0.231 0.385  1.27   168 &lt;chr [44]&gt; 
10 HALLMARK_APICAL_JUNCTI… 7.75e-2 2.47e-1   0.192 0.384  1.25   157 &lt;chr [43]&gt; 
# ℹ 40 more rows</code></pre>
</div>
</div>
<ul>
<li>Plot the normalized enrichment scores per pathway. Arrange the pathways based on NES using <code>reorder()</code>. Colour the bar indicating whether or not the gene set / pathway was significant.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fgseaResTidy, <span class="fu">aes</span>(NES, <span class="fu">reorder</span>(pathway, NES))) <span class="sc">+</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill =</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Normalized Enrichment Score"</span>, <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Hallmark pathways NES from GSEA"</span>) <span class="sc">+</span> </span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="10-pathways_files/figure-html/unnamed-chunk-30-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use plotEnrichment() to plot GSEA enrichment plot per gene set / pathway</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plotEnrichment(hallmark_list[["HALLMARK_PEROXISOME"]], ranks) +</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_y_continuous(limits = c(-0.2,0.2))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="output-the-sessioninfo-to-have-a-record-of-packages-used." class="level1">
<h1>Output the <code>sessionInfo()</code> to have a record of packages used.</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /package/R-base/4.3.0/lib/R/lib/libRblas.so 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/London
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] msigdbr_7.5.1    fgsea_1.26.0     gprofiler2_0.2.2 lubridate_1.9.2 
 [5] forcats_1.0.0    stringr_1.5.0    dplyr_1.1.3      purrr_1.0.2     
 [9] readr_2.1.4      tidyr_1.3.0      tibble_3.2.1     ggplot2_3.5.1   
[13] tidyverse_2.0.0 

loaded via a namespace (and not attached):
 [1] fastmatch_1.1-4     babelgene_22.9      gtable_0.3.4       
 [4] xfun_0.52           htmlwidgets_1.6.2   lattice_0.21-8     
 [7] tzdb_0.4.0          vctrs_0.6.3         tools_4.3.0        
[10] crosstalk_1.2.0     bitops_1.0-7        generics_0.1.3     
[13] parallel_4.3.0      fansi_1.0.4         pkgconfig_2.0.3    
[16] Matrix_1.6-5        data.table_1.14.8   lifecycle_1.0.3    
[19] farver_2.1.1        compiler_4.3.0      munsell_0.5.0      
[22] codetools_0.2-19    httpuv_1.6.11       htmltools_0.5.6    
[25] RCurl_1.98-1.12     yaml_2.3.7          lazyeval_0.2.2     
[28] plotly_4.10.2       pillar_1.9.0        later_1.3.1        
[31] crayon_1.5.2        ellipsis_0.3.2      BiocParallel_1.34.2
[34] mime_0.12           tidyselect_1.2.0    digest_0.6.33      
[37] stringi_1.7.12      labeling_0.4.3      cowplot_1.1.1      
[40] fastmap_1.1.1       grid_4.3.0          colorspace_2.1-0   
[43] cli_3.6.1           magrittr_2.0.3      utf8_1.2.3         
[46] withr_2.5.0         scales_1.3.0        promises_1.2.1     
[49] bit64_4.0.5         timechange_0.2.0    rmarkdown_2.24     
[52] httr_1.4.7          bit_4.0.5           hms_1.1.3          
[55] shiny_1.7.5         evaluate_0.21       knitr_1.43         
[58] viridisLite_0.4.2   rlang_1.1.4         Rcpp_1.0.11        
[61] xtable_1.8-4        glue_1.6.2          rstudioapi_0.15.0  
[64] vroom_1.6.3         jsonlite_1.8.7      R6_2.5.1           </code></pre>
</div>
</div>
</section>
<section id="knit-your-report-into-an-html" class="level1">
<h1>Knit your report into an HTML</h1>
<ul>
<li>This is good practise as it creates a permanent record of your plots/analysis outputs that you can share.</li>
<li>This also makes sure your script runs without error from start to finish!</li>
</ul>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>